<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Julien Roux, University of Basel, SIB">
<meta name="dcterms.date" content="2024-06-24">

<title>Hands-on: studying the long-lasting effects of Tamoxifen treatment on the CNS transcriptome across sexes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="hands_on_recount3_files/libs/clipboard/clipboard.min.js"></script>
<script src="hands_on_recount3_files/libs/quarto-html/quarto.js"></script>
<script src="hands_on_recount3_files/libs/quarto-html/popper.min.js"></script>
<script src="hands_on_recount3_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="hands_on_recount3_files/libs/quarto-html/anchor.min.js"></script>
<link href="hands_on_recount3_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="hands_on_recount3_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="hands_on_recount3_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="hands_on_recount3_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="hands_on_recount3_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Hands-on: studying the long-lasting effects of Tamoxifen treatment on the CNS transcriptome across sexes</h1>
            <p class="subtitle lead">SIB days workshop: Incorporating biological sex as a variable in the design and analysis of biomedical research experiments</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Julien Roux, University of Basel, SIB </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 24, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#loading-data-in-r" id="toc-loading-data-in-r" class="nav-link" data-scroll-target="#loading-data-in-r">Loading data in R</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data">Read in data</a></li>
  <li><a href="#examine-rangedsummarizedexperiment-object" id="toc-examine-rangedsummarizedexperiment-object" class="nav-link" data-scroll-target="#examine-rangedsummarizedexperiment-object">Examine RangedSummarizedExperiment object</a></li>
  <li><a href="#add-metadata-columns" id="toc-add-metadata-columns" class="nav-link" data-scroll-target="#add-metadata-columns">Add metadata columns</a></li>
  <li><a href="#convert-to-dgelist-object" id="toc-convert-to-dgelist-object" class="nav-link" data-scroll-target="#convert-to-dgelist-object">Convert to DGEList object</a></li>
  <li><a href="#convert-to-deseqdataset-object" id="toc-convert-to-deseqdataset-object" class="nav-link" data-scroll-target="#convert-to-deseqdataset-object">Convert to DESeqDataSet object</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>The past decades have seen large calls for consideration of the sex dimension into research and clinical projects. However as we progress towards more frequent inclusion of this variable, the design of experiments evolved to be much more complex, <a href="https://www.cell.com/cell/fulltext/S0092-8674(24)00174-0">in particular challenging the analysis steps</a>.</p>
<p>In this hands-on session we would like to focus on a particular transcriptomic dataset where the sex variable was considered in addition to other variables of interest, and illustrate a differential expression analysis aiming at rigorously testing for sex differences and notably the presence of interactions between factors. Following on Frédéric’s presentation, we would like to use this hands-on to draw your attention at power issues potentially affecting conclusions, and how to best report results of your analyses.</p>
<p>Today’s dataset is described in this paper <a href="https://link.springer.com/article/10.1007/s11357-019-00090-2">“Tamoxifen induction of Cre recombinase does not cause long-lasting or sexually divergent responses in the CNS epigenome or transcriptome: implications for the design of aging studies”</a>, and focuses on the potential side effects of Tamoxifen treatment, widely used to induce CreERT2 activity in transgenic mouse systems. Tamoxifen acts as an antagonist of estrogen receptor (ER), which could cause differences in response across sexes.</p>
<p><img src="Screenshot%202024-06-17%20at%2014.55.37.png" class="img-fluid"></p>
</section>
<section id="loading-data-in-r" class="level1">
<h1>Loading data in R</h1>
<div class="cell" data-hash="hands_on_recount3_cache/html/unnamed-chunk-1_ba6e4c7c482a2959ddc5186adbe16c3f">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">## If packages are missing: install.packages('BiocManager')</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="do">## BiocManager::install('...')</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Setting up working directory</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(recount3)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(devtools<span class="sc">::</span><span class="fu">session_info</span>(), <span class="st">"session_info_recount3.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-data" class="level1">
<h1>Read in data</h1>
<p>Raw and processed data are available from GEO <a href="GEO:%20https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE135752">(accession GSE135752)</a></p>
<p>Today we will load directly the reprocessed data from the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02533-6">recount3 project</a>, a project which reprocessed uniformly thousands of bulk RNA-seq datasets from human and mouse. A <a href="https://bioconductor.org/packages/3.17/bioc/html/recount3.html">Bioconductor package</a> allows to easily retrieve this data in the form of standard Bioconductor classes objects, such as <code>RangedSummarizedExperiment</code>.</p>
<p><img src="wrench-and-hammer.png" class="img-fluid"> Connect to the <a href="https://jhubiostatistics.shinyapps.io/recount3-study-explorer/">recount3 study explorer</a> Shiny app and search for our dataset (accession “SRP218156”).</p>
<p><img src="elemental-tip.png" class="img-fluid"> Another database of reprocessed datasets is GREIN, which provides interesting QC figures that could be helpful: http://www.ilincs.org/apps/grein/?gse=GSE135752</p>
<div class="cell" data-hash="hands_on_recount3_cache/html/unnamed-chunk-2_a903680ace57c3a4610104109ba48154">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## retrieve data frame listing all available projects in mouse</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>all_projects <span class="ot">&lt;-</span> <span class="fu">available_projects</span>(<span class="at">organism =</span> <span class="st">"mouse"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Select our dataset</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>selected <span class="ot">&lt;-</span> all_projects <span class="sc">|&gt;</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(project <span class="sc">==</span> <span class="st">"SRP218156"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Create a RangedSummarizedExperiment object at the gene level</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>rse_gene <span class="ot">&lt;-</span> <span class="fu">create_rse</span>(<span class="at">project_info =</span> selected)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="examine-rangedsummarizedexperiment-object" class="level1">
<h1>Examine RangedSummarizedExperiment object</h1>
<div class="cell" data-hash="hands_on_recount3_cache/html/unnamed-chunk-3_d92bcf564e80387d0bb1c658fde8dea2">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## The recount project records the sum of the base level coverage for each</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="do">## gene. These raw counts have to be transformed to 'classical' read-counts to</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="do">## be processed by standard DE analysis tools</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(rse_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "raw_counts"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(rse_gene, <span class="st">"counts"</span>) <span class="ot">&lt;-</span> <span class="fu">transform_counts</span>(rse_gene)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Information about our reprocessed dataset</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(rse_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$time_created
[1] "2024-06-18 15:17:18 CEST"

$recount3_version
          package ondiskversion loadedversion
recount3 recount3        1.12.0        1.12.0
                                                                                  path
recount3 /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/recount3
                                                                            loadedpath
recount3 /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/recount3
         attached is_base       date       source md5ok
recount3     TRUE   FALSE 2023-10-26 Bioconductor    NA
                                                                      library
recount3 /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library

$project
[1] "SRP218156"

$project_home
[1] "data_sources/sra"

$type
[1] "gene"

$organism
[1] "mouse"

$annotation
[1] "gencode_v23"

$recount3_url
[1] "http://duffel.rail.bio/recount3"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Gene metadata</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowRanges</span>(rse_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 55421 ranges and 11 metadata columns:
                         seqnames            ranges strand |   source     type
                            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt;
  ENSMUSG00000079800.2 GL456210.1        9124-58882      - |  ENSEMBL     gene
  ENSMUSG00000095092.1 GL456210.1     108390-110303      - |  ENSEMBL     gene
  ENSMUSG00000079192.2 GL456210.1     123792-124928      + |  ENSEMBL     gene
  ENSMUSG00000079794.2 GL456210.1     135395-136519      - |  ENSEMBL     gene
  ENSMUSG00000094799.1 GL456210.1     147792-149707      + |  ENSEMBL     gene
                   ...        ...               ...    ... .      ...      ...
  ENSMUSG00000095366.2       chrY 90752427-90755467      - |  HAVANA      gene
  ENSMUSG00000095134.2       chrY 90753057-90763485      + |  HAVANA      gene
  ENSMUSG00000096768.8       chrY 90784738-90816465      + |  HAVANA      gene
  ENSMUSG00000096850.1       chrY 90838869-90839177      - |  ENSEMBL     gene
  ENSMUSG00000099871.1       chrY 90837413-90844040      + |  HAVANA      gene
                       bp_length     phase              gene_id
                       &lt;numeric&gt; &lt;integer&gt;          &lt;character&gt;
  ENSMUSG00000079800.2      1271      &lt;NA&gt; ENSMUSG00000079800.2
  ENSMUSG00000095092.1       366      &lt;NA&gt; ENSMUSG00000095092.1
  ENSMUSG00000079192.2       255      &lt;NA&gt; ENSMUSG00000079192.2
  ENSMUSG00000079794.2       255      &lt;NA&gt; ENSMUSG00000079794.2
  ENSMUSG00000094799.1       366      &lt;NA&gt; ENSMUSG00000094799.1
                   ...       ...       ...                  ...
  ENSMUSG00000095366.2      1181      &lt;NA&gt; ENSMUSG00000095366.2
  ENSMUSG00000095134.2      1248      &lt;NA&gt; ENSMUSG00000095134.2
  ENSMUSG00000096768.8      4501      &lt;NA&gt; ENSMUSG00000096768.8
  ENSMUSG00000096850.1       309      &lt;NA&gt; ENSMUSG00000096850.1
  ENSMUSG00000099871.1       548      &lt;NA&gt; ENSMUSG00000099871.1
                                    gene_type   gene_name       level
                                  &lt;character&gt; &lt;character&gt; &lt;character&gt;
  ENSMUSG00000079800.2         protein_coding  AC125149.3           3
  ENSMUSG00000095092.1         protein_coding  AC125149.5           3
  ENSMUSG00000079192.2         protein_coding  AC125149.1           3
  ENSMUSG00000079794.2         protein_coding  AC125149.2           3
  ENSMUSG00000094799.1         protein_coding  AC125149.4           3
                   ...                    ...         ...         ...
  ENSMUSG00000095366.2                 lncRNA     Gm21860           2
  ENSMUSG00000095134.2 unprocessed_pseudogene    Mid1-ps1           2
  ENSMUSG00000096768.8                 lncRNA     Gm47283           2
  ENSMUSG00000096850.1         protein_coding     Gm21748           3
  ENSMUSG00000099871.1 unprocessed_pseudogene     Gm21742           2
                            mgi_id          havana_gene         tag
                       &lt;character&gt;          &lt;character&gt; &lt;character&gt;
  ENSMUSG00000079800.2        &lt;NA&gt;                 &lt;NA&gt;        &lt;NA&gt;
  ENSMUSG00000095092.1        &lt;NA&gt;                 &lt;NA&gt;        &lt;NA&gt;
  ENSMUSG00000079192.2        &lt;NA&gt;                 &lt;NA&gt;        &lt;NA&gt;
  ENSMUSG00000079794.2        &lt;NA&gt;                 &lt;NA&gt;        &lt;NA&gt;
  ENSMUSG00000094799.1        &lt;NA&gt;                 &lt;NA&gt;        &lt;NA&gt;
                   ...         ...                  ...         ...
  ENSMUSG00000095366.2 MGI:5434024 OTTMUSG00000074801.1        &lt;NA&gt;
  ENSMUSG00000095134.2 MGI:5780070 OTTMUSG00000047373.1        &lt;NA&gt;
  ENSMUSG00000096768.8 MGI:6096131 OTTMUSG00000074820.2        &lt;NA&gt;
  ENSMUSG00000096850.1 MGI:5433912                 &lt;NA&gt;        &lt;NA&gt;
  ENSMUSG00000099871.1 MGI:5433906 OTTMUSG00000047374.1        &lt;NA&gt;
  -------
  seqinfo: 45 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(rse_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 55421 rows and 11 columns
                       source     type bp_length     phase              gene_id
                     &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt; &lt;integer&gt;          &lt;character&gt;
ENSMUSG00000079800.2  ENSEMBL     gene      1271        NA ENSMUSG00000079800.2
ENSMUSG00000095092.1  ENSEMBL     gene       366        NA ENSMUSG00000095092.1
ENSMUSG00000079192.2  ENSEMBL     gene       255        NA ENSMUSG00000079192.2
ENSMUSG00000079794.2  ENSEMBL     gene       255        NA ENSMUSG00000079794.2
ENSMUSG00000094799.1  ENSEMBL     gene       366        NA ENSMUSG00000094799.1
...                       ...      ...       ...       ...                  ...
ENSMUSG00000095366.2  HAVANA      gene      1181        NA ENSMUSG00000095366.2
ENSMUSG00000095134.2  HAVANA      gene      1248        NA ENSMUSG00000095134.2
ENSMUSG00000096768.8  HAVANA      gene      4501        NA ENSMUSG00000096768.8
ENSMUSG00000096850.1  ENSEMBL     gene       309        NA ENSMUSG00000096850.1
ENSMUSG00000099871.1  HAVANA      gene       548        NA ENSMUSG00000099871.1
                                  gene_type   gene_name       level      mgi_id
                                &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
ENSMUSG00000079800.2         protein_coding  AC125149.3           3          NA
ENSMUSG00000095092.1         protein_coding  AC125149.5           3          NA
ENSMUSG00000079192.2         protein_coding  AC125149.1           3          NA
ENSMUSG00000079794.2         protein_coding  AC125149.2           3          NA
ENSMUSG00000094799.1         protein_coding  AC125149.4           3          NA
...                                     ...         ...         ...         ...
ENSMUSG00000095366.2                 lncRNA     Gm21860           2 MGI:5434024
ENSMUSG00000095134.2 unprocessed_pseudogene    Mid1-ps1           2 MGI:5780070
ENSMUSG00000096768.8                 lncRNA     Gm47283           2 MGI:6096131
ENSMUSG00000096850.1         protein_coding     Gm21748           3 MGI:5433912
ENSMUSG00000099871.1 unprocessed_pseudogene     Gm21742           2 MGI:5433906
                              havana_gene         tag
                              &lt;character&gt; &lt;character&gt;
ENSMUSG00000079800.2                   NA          NA
ENSMUSG00000095092.1                   NA          NA
ENSMUSG00000079192.2                   NA          NA
ENSMUSG00000079794.2                   NA          NA
ENSMUSG00000094799.1                   NA          NA
...                                   ...         ...
ENSMUSG00000095366.2 OTTMUSG00000074801.1          NA
ENSMUSG00000095134.2 OTTMUSG00000047373.1          NA
ENSMUSG00000096768.8 OTTMUSG00000074820.2          NA
ENSMUSG00000096850.1                   NA          NA
ENSMUSG00000099871.1 OTTMUSG00000047374.1          NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Parse the Ensembl IDs (useful later)</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(rse_gene)<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.</span><span class="sc">\\</span><span class="st">d+$"</span>, <span class="st">""</span>, <span class="fu">rowData</span>(rse_gene)<span class="sc">$</span>gene_id)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(rse_gene) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(rse_gene)<span class="sc">$</span>gene_id</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Sample metadata: a lot of columns!  colData(rse_gene)</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>recount3_cols <span class="ot">&lt;-</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(rse_gene))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Group columns by source</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(<span class="fu">table</span>(<span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, recount3_cols)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
      BigWigURL     external_id         rail_id           study recount_project 
              1               1               1               1               5 
   recount_pred  recount_seq_qc             sra      recount_qc 
              7              12              40             109 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## The colData from SRA include the sample information. What does each column</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="do">## represent?</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(rse_gene)[, <span class="fu">grep</span>(<span class="st">"sra"</span>, recount3_cols)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 6 rows and 40 columns
           sra.sample_acc.x sra.experiment_acc sra.submission_acc
                &lt;character&gt;        &lt;character&gt;        &lt;character&gt;
SRR9956060       SRS5258490         SRX6704419          SRA938758
SRR9956061       SRS5258491         SRX6704420          SRA938758
SRR9956062       SRS5258492         SRX6704421          SRA938758
SRR9956063       SRS5258493         SRX6704422          SRA938758
SRR9956064       SRS5258494         SRX6704423          SRA938758
SRR9956065       SRS5258495         SRX6704424          SRA938758
           sra.submission_center sra.submission_lab        sra.study_title
                     &lt;character&gt;          &lt;logical&gt;            &lt;character&gt;
SRR9956060                   GEO                 NA Tamoxifen induction ..
SRR9956061                   GEO                 NA Tamoxifen induction ..
SRR9956062                   GEO                 NA Tamoxifen induction ..
SRR9956063                   GEO                 NA Tamoxifen induction ..
SRR9956064                   GEO                 NA Tamoxifen induction ..
SRR9956065                   GEO                 NA Tamoxifen induction ..
               sra.study_abstract sra.study_description   sra.experiment_title
                      &lt;character&gt;             &lt;logical&gt;            &lt;character&gt;
SRR9956060 The systemic deliver..                    NA GSM4027456: ctx951; ..
SRR9956061 The systemic deliver..                    NA GSM4027457: ctx952; ..
SRR9956062 The systemic deliver..                    NA GSM4027458: ctx953; ..
SRR9956063 The systemic deliver..                    NA GSM4027459: ctx954; ..
SRR9956064 The systemic deliver..                    NA GSM4027460: ctx955; ..
SRR9956065 The systemic deliver..                    NA GSM4027461: ctx957; ..
           sra.design_description sra.sample_description sra.library_name
                        &lt;logical&gt;              &lt;logical&gt;        &lt;logical&gt;
SRR9956060                     NA                     NA               NA
SRR9956061                     NA                     NA               NA
SRR9956062                     NA                     NA               NA
SRR9956063                     NA                     NA               NA
SRR9956064                     NA                     NA               NA
SRR9956065                     NA                     NA               NA
           sra.library_strategy sra.library_source sra.library_selection
                    &lt;character&gt;        &lt;character&gt;           &lt;character&gt;
SRR9956060              RNA-Seq     TRANSCRIPTOMIC                  cDNA
SRR9956061              RNA-Seq     TRANSCRIPTOMIC                  cDNA
SRR9956062              RNA-Seq     TRANSCRIPTOMIC                  cDNA
SRR9956063              RNA-Seq     TRANSCRIPTOMIC                  cDNA
SRR9956064              RNA-Seq     TRANSCRIPTOMIC                  cDNA
SRR9956065              RNA-Seq     TRANSCRIPTOMIC                  cDNA
           sra.library_layout sra.paired_nominal_length
                  &lt;character&gt;                 &lt;logical&gt;
SRR9956060             paired                        NA
SRR9956061             paired                        NA
SRR9956062             paired                        NA
SRR9956063             paired                        NA
SRR9956064             paired                        NA
SRR9956065             paired                        NA
           sra.paired_nominal_stdev sra.library_construction_protocol
                          &lt;logical&gt;                       &lt;character&gt;
SRR9956060                       NA            AllPrep DNA/RNA Mini..
SRR9956061                       NA            AllPrep DNA/RNA Mini..
SRR9956062                       NA            AllPrep DNA/RNA Mini..
SRR9956063                       NA            AllPrep DNA/RNA Mini..
SRR9956064                       NA            AllPrep DNA/RNA Mini..
SRR9956065                       NA            AllPrep DNA/RNA Mini..
           sra.platform_model  sra.sample_attributes sra.experiment_attributes
                  &lt;character&gt;            &lt;character&gt;               &lt;character&gt;
SRR9956060        NextSeq 500 age;;3 Months|Sex;;f..                          
SRR9956061        NextSeq 500 age;;3 Months|Sex;;f..                          
SRR9956062        NextSeq 500 age;;3 Months|Sex;;f..                          
SRR9956063        NextSeq 500 age;;3 Months|Sex;;f..                          
SRR9956064        NextSeq 500 age;;3 Months|Sex;;f..                          
SRR9956065        NextSeq 500 age;;3 Months|Sex;;f..                          
           sra.spot_length sra.sample_name sra.sample_title sra.sample_bases
                 &lt;logical&gt;     &lt;character&gt;      &lt;character&gt;        &lt;numeric&gt;
SRR9956060              NA      GSM4027456           ctx951       1224277008
SRR9956061              NA      GSM4027457           ctx952       1636327120
SRR9956062              NA      GSM4027458           ctx953       1259195512
SRR9956063              NA      GSM4027459           ctx954       1794200552
SRR9956064              NA      GSM4027460           ctx955       1643631936
SRR9956065              NA      GSM4027461           ctx957       1361214720
           sra.sample_spots   sra.run_published  sra.size sra.run_total_bases
                  &lt;integer&gt;         &lt;character&gt; &lt;integer&gt;           &lt;numeric&gt;
SRR9956060          8054454 2019-08-14 17:36:05 481063671          1224277008
SRR9956061         10765310 2019-08-14 17:36:05 653770224          1636327120
SRR9956062          8284181 2019-08-14 17:36:05 513989033          1259195512
SRR9956063         11803951 2019-08-14 17:36:05 706586162          1794200552
SRR9956064         10813368 2019-08-14 17:36:05 646656958          1643631936
SRR9956065          8955360 2019-08-14 17:36:05 550616703          1361214720
           sra.run_total_spots sra.num_reads sra.num_spots
                     &lt;integer&gt;     &lt;integer&gt;     &lt;integer&gt;
SRR9956060             8054454             2       8054454
SRR9956061            10765310             2      10765310
SRR9956062             8284181             2       8284181
SRR9956063            11803951             2      11803951
SRR9956064            10813368             2      10813368
SRR9956065             8955360             2       8955360
                    sra.read_info sra.run_alias sra.run_center_name
                      &lt;character&gt;   &lt;character&gt;           &lt;logical&gt;
SRR9956060 index:0,count:805445.. GSM4027456_r1                  NA
SRR9956061 index:0,count:107653.. GSM4027457_r1                  NA
SRR9956062 index:0,count:828418.. GSM4027458_r1                  NA
SRR9956063 index:0,count:118039.. GSM4027459_r1                  NA
SRR9956064 index:0,count:108133.. GSM4027460_r1                  NA
SRR9956065 index:0,count:895536.. GSM4027461_r1                  NA
           sra.run_broker_name sra.run_center  sra.mesa sra.pubmed_id
                     &lt;logical&gt;      &lt;logical&gt; &lt;logical&gt;     &lt;logical&gt;
SRR9956060                  NA             NA        NA            NA
SRR9956061                  NA             NA        NA            NA
SRR9956062                  NA             NA        NA            NA
SRR9956063                  NA             NA        NA            NA
SRR9956064                  NA             NA        NA            NA
SRR9956065                  NA             NA        NA            NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This column corresponds to the sample name on GEO</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(rse_gene<span class="sc">$</span>sra.sample_title)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   951    952    953    954    955    957    958    959    960    961    962 
     2      2      2      2      2      2      2      2      2      2      2 
   963    964    965    966 ctx951 ctx952 ctx953 ctx954 ctx955 ctx957 ctx958 
     2      2      2      2      1      1      1      1      1      1      1 
ctx959 ctx960 ctx961 ctx962 ctx963 ctx964 ctx965 ctx966 ret951 ret952 ret953 
     1      1      1      1      1      1      1      1      1      1      1 
ret954 ret955 ret957 ret958 ret959 ret960 ret961 ret962 ret963 ret964 ret965 
     1      1      1      1      1      1      1      1      1      1      1 
ret966 
     1 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">## For some samples we have multiple runs (SRR IDs), let's sum their read</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="do">## counts</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>rse_gene <span class="ot">&lt;-</span> <span class="fu">aggregateAcrossCells</span>(rse_gene, <span class="at">ids =</span> rse_gene<span class="sc">$</span>sra.sample_title, <span class="at">statistics =</span> <span class="st">"sum"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="add-metadata-columns" class="level1">
<h1>Add metadata columns</h1>
<div class="cell" data-hash="hands_on_recount3_cache/html/unnamed-chunk-4_9237cb9e56ae21d62e0d32586e628da6">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Simplify metadata columns of interest by parsing 'sra.sample_attributes'</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>rse_gene<span class="sc">$</span>sex <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">strsplit2</span>(rse_gene<span class="sc">$</span>sra.sample_attributes, <span class="at">split =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>)[,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="dv">2</span>])</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(rse_gene<span class="sc">$</span>sex) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Sex;;"</span>, <span class="st">""</span>, <span class="fu">levels</span>(rse_gene<span class="sc">$</span>sex))</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>rse_gene<span class="sc">$</span>tissue <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">strsplit2</span>(rse_gene<span class="sc">$</span>sra.sample_attributes, <span class="at">split =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>)[,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    <span class="dv">3</span>])</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(rse_gene<span class="sc">$</span>tissue) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"source_name;;"</span>, <span class="st">""</span>, <span class="fu">levels</span>(rse_gene<span class="sc">$</span>tissue))</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>rse_gene<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">strsplit2</span>(rse_gene<span class="sc">$</span>sra.sample_attributes, <span class="at">split =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>)[,</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="dv">4</span>])</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(rse_gene<span class="sc">$</span>treatment) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"treatment;;"</span>, <span class="st">""</span>, <span class="fu">levels</span>(rse_gene<span class="sc">$</span>treatment))</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(rse_gene<span class="sc">$</span>treatment) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s</span><span class="sc">\\</span><span class="st">(Sunflower</span><span class="sc">\\</span><span class="st">sOil</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">""</span>, <span class="fu">levels</span>(rse_gene<span class="sc">$</span>treatment))</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Relevel to have Vehicle as reference</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>rse_gene<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">relevel</span>(rse_gene<span class="sc">$</span>treatment, <span class="at">ref =</span> <span class="st">"Vehicle"</span>)</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="do">## From the methods it seems that the 3 tissues were extracted from the same</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="do">## mouse</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>rse_gene<span class="sc">$</span>mouse <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">gsub</span>(<span class="st">"ctx|ret"</span>, <span class="st">""</span>, rse_gene<span class="sc">$</span>sra.sample_title))</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Add grouping variable (grouping 2 factors together, will be useful later)</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a><span class="do">## and order levels in a logical way</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>rse_gene<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">paste0</span>(rse_gene<span class="sc">$</span>treatment, <span class="st">"."</span>, rse_gene<span class="sc">$</span>sex), <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Vehicle.female"</span>,</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tamoxifen.female"</span>, <span class="st">"Vehicle.male"</span>, <span class="st">"Tamoxifen.male"</span>))</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="do">## Overview of experiment design</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(rse_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 55421    45</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(rse_gene<span class="sc">$</span>sex, rse_gene<span class="sc">$</span>treatment, rse_gene<span class="sc">$</span>tissue)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>, ,  = Cortex

        
         Vehicle Tamoxifen
  female       3         4
  male         4         4

, ,  = Hippocampus

        
         Vehicle Tamoxifen
  female       3         4
  male         4         4

, ,  = Retina

        
         Vehicle Tamoxifen
  female       3         4
  male         4         4</code></pre>
</div>
</div>
</section>
<section id="convert-to-dgelist-object" class="level1">
<h1>Convert to DGEList object</h1>
<div class="cell" data-hash="hands_on_recount3_cache/html/unnamed-chunk-5_84c9e99f35d735dd0a496802e9bd18a6">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">SE2DGEList</span>(rse_gene)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dge, <span class="st">"DGEList.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="convert-to-deseqdataset-object" class="level1">
<h1>Convert to DESeqDataSet object</h1>
<div class="cell" data-hash="hands_on_recount3_cache/html/unnamed-chunk-6_98de9d73c4576fc63c828fe6166a2151">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSet</span>(rse_gene, <span class="at">design =</span> <span class="sc">~</span><span class="dv">1</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dds, <span class="st">"DESeqDataSet.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>