<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Julien Roux, University of Basel, SIB">
<meta name="author" content="Leonore Wigger, Vital-IT, SIB">
<meta name="dcterms.date" content="2024-06-24">

<title>Hands-on: studying the long-lasting effects of Tamoxifen treatment on the CNS transcriptome across sexes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="hands_on_limma_advanced_files/libs/clipboard/clipboard.min.js"></script>
<script src="hands_on_limma_advanced_files/libs/quarto-html/quarto.js"></script>
<script src="hands_on_limma_advanced_files/libs/quarto-html/popper.min.js"></script>
<script src="hands_on_limma_advanced_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="hands_on_limma_advanced_files/libs/quarto-html/anchor.min.js"></script>
<link href="hands_on_limma_advanced_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="hands_on_limma_advanced_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="hands_on_limma_advanced_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="hands_on_limma_advanced_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="hands_on_limma_advanced_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Hands-on: studying the long-lasting effects of Tamoxifen treatment on the CNS transcriptome across sexes</h1>
            <p class="subtitle lead">SIB days workshop: Incorporating biological sex as a variable in the design and analysis of biomedical research experiments</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Authors</div>
      <div class="quarto-title-meta-contents">
               <p>Julien Roux, University of Basel, SIB </p>
               <p>Leonore Wigger, Vital-IT, SIB </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 24, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#loading-data-in-r" id="toc-loading-data-in-r" class="nav-link" data-scroll-target="#loading-data-in-r">Loading data in R</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data">Read in data</a></li>
  <li><a href="#gene-filtering" id="toc-gene-filtering" class="nav-link" data-scroll-target="#gene-filtering">Gene filtering</a></li>
  <li><a href="#qc-checks" id="toc-qc-checks" class="nav-link" data-scroll-target="#qc-checks">QC checks</a>
  <ul class="collapse">
  <li><a href="#expression-of-xist-gene" id="toc-expression-of-xist-gene" class="nav-link" data-scroll-target="#expression-of-xist-gene">Expression of <em>Xist</em> gene</a></li>
  <li><a href="#expression-of-y-chromosome-genes" id="toc-expression-of-y-chromosome-genes" class="nav-link" data-scroll-target="#expression-of-y-chromosome-genes">Expression of Y-chromosome genes</a></li>
  <li><a href="#principal-component-analysis" id="toc-principal-component-analysis" class="nav-link" data-scroll-target="#principal-component-analysis">Principal component analysis</a></li>
  </ul></li>
  <li><a href="#differential-expression-analysis" id="toc-differential-expression-analysis" class="nav-link" data-scroll-target="#differential-expression-analysis">Differential expression analysis</a>
  <ul class="collapse">
  <li><a href="#interactions-term" id="toc-interactions-term" class="nav-link" data-scroll-target="#interactions-term">Interactions term</a></li>
  </ul></li>
  <li><a href="#gsea" id="toc-gsea" class="nav-link" data-scroll-target="#gsea">GSEA</a></li>
  <li><a href="#to-finish" id="toc-to-finish" class="nav-link" data-scroll-target="#to-finish">To finish</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>The past decades have seen large calls for consideration of the sex dimension into research and clinical projects. However as we progress towards more frequent inclusion of this variable, the design of experiments evolved to be much more complex, <a href="https://www.cell.com/cell/fulltext/S0092-8674(24)00174-0">in particular challenging the analysis steps</a>.</p>
<p>In this hands-on session we would like to focus on a particular transcriptomic dataset where the sex variable was considered in addition to other variables of interest, and illustrate a differential expression analysis aiming at rigorously testing for sex differences and notably the presence of interactions between factors. Following on Frédéric’s presentation, we would like to use this hands-on to draw your attention at power issues potentially affecting conclusions, and how to best report results of your analyses.</p>
<p>Today’s dataset is described in this paper <a href="https://link.springer.com/article/10.1007/s11357-019-00090-2">“Tamoxifen induction of Cre recombinase does not cause long-lasting or sexually divergent responses in the CNS epigenome or transcriptome: implications for the design of aging studies”</a>, and focuses on the potential side effects of Tamoxifen treatment, widely used to induce CreERT2 activity in transgenic mouse systems. Tamoxifen acts as an antagonist of estrogen receptor (ER), which could cause differences in response across sexes.</p>
<p><img src="Screenshot%202024-06-17%20at%2014.55.37.png" class="img-fluid"></p>
<p><strong>This script presents a slightly more complex way – but maybe better – of analyzing the dataset. Use it if you are already familiar with DE analysis with the limma/edgeR packages</strong></p>
</section>
<section id="loading-data-in-r" class="level1">
<h1>Loading data in R</h1>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-1_cd8ab69fd8c3724d6c8eee165a06af30">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">## If packages are missing: install.packages('BiocManager')</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="do">## BiocManager::install('...')</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Setting up working directory</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="do">## color palettes</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>myPalette <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="do">## DE analysis</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Needed for GSEA</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(devtools<span class="sc">::</span><span class="fu">session_info</span>(), <span class="st">"session_info_limma_advanced.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-data" class="level1">
<h1>Read in data</h1>
<p>Raw and processed data are available from GEO <a href="GEO:%20https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE135752">(accession GSE135752)</a></p>
<p>Today we will load directly the reprocessed data loaded from the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02533-6">recount3 project</a>, a project which reprocessed uniformly thousands of bulk RNA-seq datasets from human and mouse. See the <code>hands_on_recount3.qmd</code> script where we import and format the <code>DGEList</code> object.</p>
</section>
<section id="gene-filtering" class="level1">
<h1>Gene filtering</h1>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-2_377b81889a086b67fa34b87da811adf6">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load DESeqDataSet object</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"DGEList.rds"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Or load directly from Github dge &lt;-</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="do">## readRDS(url('https://raw.githubusercontent.com/julien-roux/SIB_days_2024_workshop_EDI/main/DGEList.rds'))</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Visualize distribution before filtering (protein-coding genes only)</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(edgeR<span class="sc">::</span><span class="fu">cpm</span>(dge, <span class="at">log =</span> T)[dge<span class="sc">$</span>genes<span class="sc">$</span>gene_type <span class="sc">%in%</span> <span class="st">"protein_coding"</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    ], <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>tissue, <span class="at">col =</span> myPalette, <span class="at">legend =</span> <span class="st">"topright"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## One cortex sample is weird: let's remove it</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[, <span class="fu">colnames</span>(dge) <span class="sc">!=</span> <span class="st">"ctx962"</span>]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter lowly expressed genes</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(dge, <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>group)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>keep
FALSE  TRUE 
32710 22711 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## subset DGEList, while updating total counts</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[keep, , keep.lib.sizes <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="do">## TMM normalization</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Calculate normalized log2(CPM) values</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>log2cpm <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(dge, <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">normalized.lib.sizes =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(log2cpm, <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="at">col =</span> myPalette, <span class="at">legend =</span> <span class="st">"topright"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-2-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(log2cpm, <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>tissue, <span class="at">col =</span> myPalette, <span class="at">legend =</span> <span class="st">"topright"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-2-3.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="qc-checks" class="level1">
<h1>QC checks</h1>
<section id="expression-of-xist-gene" class="level2">
<h2 class="anchored" data-anchor-id="expression-of-xist-gene">Expression of <em>Xist</em> gene</h2>
<p>Xist is a lncRNA regulating the X-chromosome inactivation process in mammals, used to equalize the dosage of X-linked genes between female (XX) and male (XY). It should thus be expressed only in female samples</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-3_16a35a96aeb8bfa9a0845bfc2817ed81">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm[<span class="fu">grep</span>(<span class="st">"Xist"</span>, dge<span class="sc">$</span>genes<span class="sc">$</span>gene_name), , drop <span class="ot">=</span> <span class="cn">FALSE</span>] <span class="sc">|&gt;</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">colour =</span> tissue, <span class="at">group =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="expression-of-y-chromosome-genes" class="level2">
<h2 class="anchored" data-anchor-id="expression-of-y-chromosome-genes">Expression of Y-chromosome genes</h2>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-4_4717eb9dc4abaaab616abec073cdbff6">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Chromosome information</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(dge<span class="sc">$</span>genes<span class="sc">$</span>seqnames)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "chr1"       "chr2"       "chr3"       "chr4"       "chr5"      
 [6] "chr6"       "chr7"       "chr8"       "chr9"       "chr10"     
[11] "chr11"      "chr12"      "chr13"      "chr14"      "chr15"     
[16] "chr16"      "chr17"      "chr18"      "chr19"      "chrX"      
[21] "chrY"       "chrM"       "GL456210.1" "GL456211.1" "GL456212.1"
[26] "GL456216.1" "GL456219.1" "GL456221.1" "GL456233.1" "GL456239.1"
[31] "GL456350.1" "GL456354.1" "GL456372.1" "GL456381.1" "GL456385.1"
[36] "JH584292.1" "JH584293.1" "JH584294.1" "JH584295.1" "JH584296.1"
[41] "JH584297.1" "JH584298.1" "JH584299.1" "JH584303.1" "JH584304.1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## How many genes are left in our dataset after filtering?</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dge<span class="sc">$</span>genes<span class="sc">$</span>seqnames <span class="sc">==</span> <span class="st">"chrY"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
22700    11 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm[dge<span class="sc">$</span>genes<span class="sc">$</span>seqnames <span class="sc">==</span> <span class="st">"chrY"</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] <span class="sc">|&gt;</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">colour =</span> tissue, <span class="at">group =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">4</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> What do you observe? What is going on for some genes? (e.g., <em>Gm20775</em>)</p>
<p><img src="elemental-tip.png" class="img-fluid"> Bigwig files are available through recount3 for some of the samples and allow to visualize the read coverage (for example on the Y-chromosome) on a genome browser. For example you can provide these links to the UCSC genome browser directly:</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-5_5c9a5cf3a7fe6aafb00c829a620b7b47">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(dge<span class="sc">$</span>samples<span class="sc">$</span>BigWigURL)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/14/sra.base_sums.SRP218156_SRR9956114.ALL.bw"
[2] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/15/sra.base_sums.SRP218156_SRR9956115.ALL.bw"
[3] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/16/sra.base_sums.SRP218156_SRR9956116.ALL.bw"
[4] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/17/sra.base_sums.SRP218156_SRR9956117.ALL.bw"
[5] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/18/sra.base_sums.SRP218156_SRR9956118.ALL.bw"
[6] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/19/sra.base_sums.SRP218156_SRR9956119.ALL.bw"</code></pre>
</div>
</div>
</section>
<section id="principal-component-analysis" class="level2">
<h2 class="anchored" data-anchor-id="principal-component-analysis">Principal component analysis</h2>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/pca_aae33afeb4714bfeb432fcd0b9af7b82">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">## PCA on top 500 most variable genes</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>iqrs <span class="ot">&lt;-</span> <span class="fu">apply</span>(log2cpm, <span class="dv">1</span>, IQR)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>sel <span class="ot">&lt;-</span> iqrs <span class="sc">&gt;=</span> <span class="fu">sort</span>(iqrs, <span class="at">decreasing =</span> T)[<span class="dv">500</span>]</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>pca1 <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(log2cpm[sel, ]), <span class="at">scale =</span> T)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca1)<span class="sc">$</span>importance[, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                            PC1      PC2      PC3      PC4       PC5       PC6
Standard deviation     21.49999 3.859032 2.863773 1.589076 0.9230414 0.8300593
Proportion of Variance  0.92450 0.029780 0.016400 0.005050 0.0017000 0.0013800
Cumulative Proportion   0.92450 0.954280 0.970690 0.975740 0.9774400 0.9788200
                             PC7       PC8       PC9      PC10
Standard deviation     0.8055562 0.7751416 0.7446052 0.6975254
Proportion of Variance 0.0013000 0.0012000 0.0011100 0.0009700
Cumulative Proportion  0.9801200 0.9813200 0.9824300 0.9834000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/pca-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Which sample metadata associate to which PCs?  Let's use the</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="do">## getExplanatoryPCs from scater package, which takes as input a</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="do">## SingleCellExperiment object containing dimensionality reduction results</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="at">logcounts =</span> log2cpm), <span class="at">reducedDims =</span> <span class="fu">list</span>(<span class="at">PCA =</span> pca1<span class="sc">$</span>x),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">colData =</span> dge<span class="sc">$</span>samples)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">&lt;-</span> <span class="fu">getExplanatoryPCs</span>(sce, <span class="at">n_dimred =</span> <span class="dv">10</span>, <span class="at">dimred =</span> <span class="st">"PCA"</span>, <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"sex"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"treatment"</span>, <span class="st">"tissue"</span>, <span class="st">"mouse"</span>))</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>percentVar</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              sex  treatment      tissue     mouse
PC1   0.006495126 0.01727742 99.93384555  0.503909
PC2   0.517791186 0.10868961 97.52726886  3.283735
PC3  97.542199169 1.22935290  0.25768486 97.896744
PC4   0.161111059 0.46281290  0.79569261 42.837992
PC5   0.676045162 0.13875380  0.04733970 26.723407
PC6   0.005913613 1.57780237  0.01936682 25.132588
PC7   0.251453219 0.27859827  0.02099720 34.903531
PC8   0.015834211 0.99585626  0.16509570 38.492577
PC9   0.001790200 0.54427412  0.06954958 35.126040
PC10  0.227016205 3.34983920  0.04426769 27.734207</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotExplanatoryPCs</span>(percentVar<span class="sc">/</span><span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/pca-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> To which principal components contribute which factors? Does this corresponds to the expectations? What do you conclude on the relative effect size of different factors?</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-6_000cf8615f0a11c35281f1ac776f9399">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Visualize some of these associations, for example tissues with PCs 1 and 2,</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="do">## sex with PC3</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca1<span class="sc">$</span>x[, <span class="dv">1</span>], pca1<span class="sc">$</span>x[, <span class="dv">2</span>], <span class="at">pch =</span> <span class="fu">c</span>(<span class="dv">15</span><span class="sc">:</span><span class="dv">16</span>)[dge<span class="sc">$</span>samples<span class="sc">$</span>treatment], <span class="at">col =</span> myPalette[dge<span class="sc">$</span>samples<span class="sc">$</span>tissue])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca1<span class="sc">$</span>x[, <span class="dv">2</span>], pca1<span class="sc">$</span>x[, <span class="dv">3</span>], <span class="at">pch =</span> <span class="fu">c</span>(<span class="dv">15</span><span class="sc">:</span><span class="dv">16</span>)[dge<span class="sc">$</span>samples<span class="sc">$</span>treatment], <span class="at">col =</span> myPalette[dge<span class="sc">$</span>samples<span class="sc">$</span>sex])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis" class="level1">
<h1>Differential expression analysis</h1>
<p>Here we start to differ from the <code>hands_on_limma.qmd</code> script…</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/de_analysis_9a145bf8609d15b10870c893734c6d28">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Let's update the grouping variable to include the 3 experimental factors (and order levels in a logical way)</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>dge<span class="sc">$</span>samples<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">paste0</span>(dge<span class="sc">$</span>samples<span class="sc">$</span>tissue, <span class="st">"."</span>, dge<span class="sc">$</span>samples<span class="sc">$</span>treatment, <span class="st">"."</span>, dge<span class="sc">$</span>samples<span class="sc">$</span>sex), </span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Cortex.Vehicle.female"</span>, <span class="st">"Cortex.Tamoxifen.female"</span>, </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Cortex.Vehicle.male"</span>, <span class="st">"Cortex.Tamoxifen.male"</span>,</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Hippocampus.Vehicle.female"</span>, <span class="st">"Hippocampus.Tamoxifen.female"</span>, </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Hippocampus.Vehicle.male"</span>, <span class="st">"Hippocampus.Tamoxifen.male"</span>,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Retina.Vehicle.female"</span>, <span class="st">"Retina.Tamoxifen.female"</span>, </span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Retina.Vehicle.male"</span>, <span class="st">"Retina.Tamoxifen.male"</span>))</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Design matrix</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>moma <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> group, <span class="at">data=</span>dge<span class="sc">$</span>samples) </span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(moma) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"group"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(moma)) <span class="do">## Easier to manipulate contrasts</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Contrast matrix: </span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="do">## - Direct male vs. female comparison (in controls only), in each tissue</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a><span class="do">## - effect of Tamoxifen treatment in each sex separately, in each tissue</span></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a><span class="do">## - interaction term to extract genes reacting differently to Tamoxifen treatment in male and female, in each tissue</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>contrasts.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">Male_vs_female_Cortex         =</span> Cortex.Vehicle.male <span class="sc">-</span> Cortex.Vehicle.female, </span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tamoxifen_effet_male_Cortex   =</span> Cortex.Tamoxifen.male <span class="sc">-</span> Cortex.Vehicle.male,</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tamoxifen_effet_female_Cortex =</span> Cortex.Tamoxifen.female <span class="sc">-</span> Cortex.Vehicle.female,</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">Interaction_Cortex            =</span> (Cortex.Tamoxifen.male <span class="sc">-</span> Cortex.Vehicle.male) <span class="sc">-</span> (Cortex.Tamoxifen.female <span class="sc">-</span> Cortex.Vehicle.female),</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">Male_vs_female_Hippocampus         =</span> Hippocampus.Vehicle.male <span class="sc">-</span> Hippocampus.Vehicle.female, </span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tamoxifen_effet_male_Hippocampus   =</span> Hippocampus.Tamoxifen.male <span class="sc">-</span> Hippocampus.Vehicle.male,</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tamoxifen_effet_female_Hippocampus =</span> Hippocampus.Tamoxifen.female <span class="sc">-</span> Hippocampus.Vehicle.female,</span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">Interaction_Hippocampus            =</span> (Hippocampus.Tamoxifen.male <span class="sc">-</span> Hippocampus.Vehicle.male) <span class="sc">-</span> (Hippocampus.Tamoxifen.female <span class="sc">-</span> Hippocampus.Vehicle.female),</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">Male_vs_female_Retina         =</span> Retina.Vehicle.male <span class="sc">-</span> Retina.Vehicle.female, </span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tamoxifen_effet_male_Retina   =</span> Retina.Tamoxifen.male <span class="sc">-</span> Retina.Vehicle.male,</span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tamoxifen_effet_female_Retina =</span> Retina.Tamoxifen.female <span class="sc">-</span> Retina.Vehicle.female,</span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">Interaction_Retina            =</span> (Retina.Tamoxifen.male <span class="sc">-</span> Retina.Vehicle.male) <span class="sc">-</span> (Retina.Tamoxifen.female <span class="sc">-</span> Retina.Vehicle.female),</span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">levels=</span>moma</span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Run DE analysis</span></span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, moma, <span class="at">plot=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/de_analysis-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, moma)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, contrasts.matrix)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Extract genes significant at FDR 10%</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(fit2, <span class="at">p.value =</span> <span class="fl">0.1</span>) </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(de, <span class="dv">2</span>, table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$Male_vs_female_Cortex

   -1     0     1 
    9 22696     6 

$Tamoxifen_effet_male_Cortex

   -1     0     1 
    2 22706     3 

$Tamoxifen_effet_female_Cortex

   -1     0     1 
   26 22673    12 

$Interaction_Cortex

   -1     0     1 
    1 22709     1 

$Male_vs_female_Hippocampus

   -1     0     1 
 1120 20754   837 

$Tamoxifen_effet_male_Hippocampus

   -1     0     1 
   37 22611    63 

$Tamoxifen_effet_female_Hippocampus

   -1     0     1 
   16 22692     3 

$Interaction_Hippocampus

    0 
22711 

$Male_vs_female_Retina

   -1     0     1 
   19 22673    19 

$Tamoxifen_effet_male_Retina

   -1     0     1 
    2 22708     1 

$Tamoxifen_effet_female_Retina

   -1     0     1 
   19 22677    15 

$Interaction_Retina

   -1     0 
    1 22710 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Top genes for each contrast</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Male_vs_female_Cortex"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand  source type
ENSMUSG00000035150     chrX  94188707  94212862  24156      -  HAVANA gene
ENSMUSG00000069045     chrY   1260771   1286629  25859      -  HAVANA gene
ENSMUSG00000068457     chrY   1096861   1245759 148899      -  HAVANA gene
ENSMUSG00000056673     chrY    897788    956786  58999      +  HAVANA gene
ENSMUSG00000099312     chrX 103461218 103461285     68      - ENSEMBL gene
ENSMUSG00000069049     chrY   1010543   1028847  18305      +  HAVANA gene
ENSMUSG00000086503     chrX 103460366 103483254  22889      -  HAVANA gene
ENSMUSG00000098262     chrX 103480946 103481026     81      + ENSEMBL gene
ENSMUSG00000099876     chrY   1048393   1049134    742      +  HAVANA gene
ENSMUSG00000098743     chrX 103469671 103469796    126      - ENSEMBL gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000035150      3812    NA ENSMUSG00000035150 protein_coding   Eif2s3x
ENSMUSG00000069045      5199    NA ENSMUSG00000069045 protein_coding     Ddx3y
ENSMUSG00000068457      9791    NA ENSMUSG00000068457 protein_coding       Uty
ENSMUSG00000056673     15164    NA ENSMUSG00000056673 protein_coding     Kdm5d
ENSMUSG00000099312        68    NA ENSMUSG00000099312       misc_RNA   Gm27733
ENSMUSG00000069049      4161    NA ENSMUSG00000069049 protein_coding   Eif2s3y
ENSMUSG00000086503     17946    NA ENSMUSG00000086503         lncRNA      Xist
ENSMUSG00000098262        81    NA ENSMUSG00000098262       misc_RNA   Gm27520
ENSMUSG00000099876       742    NA ENSMUSG00000099876         lncRNA   Gm29650
ENSMUSG00000098743       126    NA ENSMUSG00000098743       misc_RNA   Gm27927
                   level      mgi_id          havana_gene               tag
ENSMUSG00000035150     2 MGI:1349431 OTTMUSG00000018084.1              &lt;NA&gt;
ENSMUSG00000069045     2 MGI:1349406 OTTMUSG00000045279.1 overlapping_locus
ENSMUSG00000068457     2  MGI:894810 OTTMUSG00000032814.5              &lt;NA&gt;
ENSMUSG00000056673     2   MGI:99780 OTTMUSG00000045277.1              &lt;NA&gt;
ENSMUSG00000099312     3 MGI:5531115                 &lt;NA&gt;              &lt;NA&gt;
ENSMUSG00000069049     2 MGI:1349430 OTTMUSG00000021559.1              &lt;NA&gt;
ENSMUSG00000086503     2   MGI:98974 OTTMUSG00000018295.5        ncRNA_host
ENSMUSG00000098262     3 MGI:5530902                 &lt;NA&gt;              &lt;NA&gt;
ENSMUSG00000099876     2 MGI:5580356 OTTMUSG00000045278.1              &lt;NA&gt;
ENSMUSG00000098743     3 MGI:5531309                 &lt;NA&gt;              &lt;NA&gt;
                        logFC    AveExpr         t      P.Value    adj.P.Val
ENSMUSG00000035150 -0.6153907  6.2281911 -8.607408 1.168905e-10 1.879204e-06
ENSMUSG00000069045 11.8563369  0.2849525  8.440435 1.952816e-10 1.879204e-06
ENSMUSG00000068457 10.0639577 -0.7664206  8.362720 2.482326e-10 1.879204e-06
ENSMUSG00000056673  9.0963566 -0.4319924  8.226329 3.788084e-10 2.150780e-06
ENSMUSG00000099312 -7.0145525 -2.8949791 -6.998680 1.842197e-08 8.367629e-05
ENSMUSG00000069049 11.0174448 -0.4760528  6.708795 4.686356e-08 1.773864e-04
ENSMUSG00000086503 -9.6058045  3.6571905 -6.603225 6.591090e-08 2.138432e-04
ENSMUSG00000098262 -6.0541090 -3.2312004 -6.383101 1.343951e-07 3.815309e-04
ENSMUSG00000099876  5.8644487 -3.1854202  5.711936 1.184688e-06 2.989493e-03
ENSMUSG00000098743 -7.5651694 -2.2759046 -5.404825 3.197897e-06 7.262744e-03
                            B
ENSMUSG00000035150 14.1858738
ENSMUSG00000069045  1.8193169
ENSMUSG00000068457  1.7525794
ENSMUSG00000056673  2.0205495
ENSMUSG00000099312  1.5220166
ENSMUSG00000069049  0.4848969
ENSMUSG00000086503  6.4897576
ENSMUSG00000098262  0.7151574
ENSMUSG00000099876 -0.6213092
ENSMUSG00000098743 -0.1002527</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Tamoxifen_effet_male_Cortex"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000022602    chr15  74669083  74672570   3488      - HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000008090     chr5 108692382 108706924  14543      + HAVANA gene
ENSMUSG00000030199     chr6 134035700 134270158 234459      + HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000042216     chr5 113243220 113310786  67567      - HAVANA gene
ENSMUSG00000037112     chr9  50892801  51009073 116273      - HAVANA gene
ENSMUSG00000026773     chr2  11471433  11554077  82645      - HAVANA gene
ENSMUSG00000086610     chr5 149006962 149012776   5815      + HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000022602      3057    NA ENSMUSG00000022602 protein_coding       Arc
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000008090      2782    NA ENSMUSG00000008090 protein_coding    Fgfrl1
ENSMUSG00000030199      5983    NA ENSMUSG00000030199 protein_coding      Etv6
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000042216      7466    NA ENSMUSG00000042216 protein_coding     Sgsm1
ENSMUSG00000037112      9539    NA ENSMUSG00000037112 protein_coding      Sik2
ENSMUSG00000026773      7561    NA ENSMUSG00000026773 protein_coding    Pfkfb3
ENSMUSG00000086610      1379    NA ENSMUSG00000086610         lncRNA   Gm15408
                   level      mgi_id          havana_gene               tag
ENSMUSG00000022602     2   MGI:88067 OTTMUSG00000016386.3              &lt;NA&gt;
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
ENSMUSG00000008090     2 MGI:2150920 OTTMUSG00000054022.2              &lt;NA&gt;
ENSMUSG00000030199     2  MGI:109336 OTTMUSG00000025993.7              &lt;NA&gt;
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3              &lt;NA&gt;
ENSMUSG00000042216     2  MGI:107320 OTTMUSG00000025828.2              &lt;NA&gt;
ENSMUSG00000037112     2 MGI:2445031 OTTMUSG00000042049.4              &lt;NA&gt;
ENSMUSG00000026773     2 MGI:2181202 OTTMUSG00000011314.6 overlapping_locus
ENSMUSG00000086610     2 MGI:3705101 OTTMUSG00000021225.1              &lt;NA&gt;
                        logFC   AveExpr         t      P.Value    adj.P.Val
ENSMUSG00000022602  0.5678203  5.636934  6.781845 3.702287e-08 0.0008408263
ENSMUSG00000066687 -0.8473846  4.422495 -6.406379 1.246328e-07 0.0014152681
ENSMUSG00000021250  0.6957341  4.877409  6.234880 2.172963e-07 0.0016450056
ENSMUSG00000008090  0.5472155  3.320005  4.901246 1.606908e-05 0.0882389377
ENSMUSG00000030199 -0.4529419  4.306002 -4.841529 1.942648e-05 0.0882389377
ENSMUSG00000037868  0.7913071  0.373090  4.732882 2.740327e-05 0.1037259411
ENSMUSG00000042216 -0.4250538  5.918650 -4.394360 7.912479e-05 0.2567147327
ENSMUSG00000037112 -0.4191221  5.271596 -4.194388 1.465715e-04 0.4160982662
ENSMUSG00000026773 -0.4002472  5.475579 -4.060708 2.202516e-04 0.5557927734
ENSMUSG00000086610  2.6136483 -1.200616  3.923225 3.333129e-04 0.7569868504
                            B
ENSMUSG00000022602  7.7699626
ENSMUSG00000066687  5.5696775
ENSMUSG00000021250  5.3990485
ENSMUSG00000008090  1.0954698
ENSMUSG00000030199  1.6268752
ENSMUSG00000037868  0.8368334
ENSMUSG00000042216  1.2253373
ENSMUSG00000037112  0.4700520
ENSMUSG00000026773  0.1493870
ENSMUSG00000086610 -4.2123568</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Tamoxifen_effet_female_Cortex"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000032846    chr13 107724617 107890064 165448      - HAVANA gene
ENSMUSG00000053414    chr16  90386013  90499553 113541      + HAVANA gene
ENSMUSG00000020423     chr1 134075170 134079120   3951      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000024222    chr17  28399094  28517527 118434      - HAVANA gene
ENSMUSG00000037112     chr9  50892801  51009073 116273      - HAVANA gene
ENSMUSG00000023034    chr15 101254269 101274795  20527      + HAVANA gene
ENSMUSG00000032402     chr9  63646767  63757994 111228      - HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000032846      5550    NA ENSMUSG00000032846 protein_coding    Zswim6
ENSMUSG00000053414      8402    NA ENSMUSG00000053414 protein_coding      Hunk
ENSMUSG00000020423      2743    NA ENSMUSG00000020423 protein_coding      Btg2
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000024222      9468    NA ENSMUSG00000024222 protein_coding     Fkbp5
ENSMUSG00000037112      9539    NA ENSMUSG00000037112 protein_coding      Sik2
ENSMUSG00000023034      3509    NA ENSMUSG00000023034 protein_coding     Nr4a1
ENSMUSG00000032402      5542    NA ENSMUSG00000032402 protein_coding     Smad3
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
                   level      mgi_id          havana_gene
ENSMUSG00000032846     2 MGI:1914513 OTTMUSG00000067880.2
ENSMUSG00000053414     2 MGI:1347352 OTTMUSG00000071848.1
ENSMUSG00000020423     2  MGI:108384 OTTMUSG00000034725.1
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1
ENSMUSG00000024222     2  MGI:104670 OTTMUSG00000024255.3
ENSMUSG00000037112     2 MGI:2445031 OTTMUSG00000042049.4
ENSMUSG00000023034     2 MGI:1352454 OTTMUSG00000071152.1
ENSMUSG00000032402     2 MGI:1201674 OTTMUSG00000021500.2
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1
                                      tag      logFC  AveExpr         t
ENSMUSG00000032846 reference_genome_error -0.7010511 5.183653 -7.820235
ENSMUSG00000053414                   &lt;NA&gt;  0.8349069 5.222101  7.610631
ENSMUSG00000020423                   &lt;NA&gt;  0.8827455 4.872284  7.302754
ENSMUSG00000021250                   &lt;NA&gt;  0.7809225 4.877409  7.164634
ENSMUSG00000024222                   &lt;NA&gt; -0.7507339 5.352605 -7.036765
ENSMUSG00000037112                   &lt;NA&gt; -0.7050311 5.271596 -6.849409
ENSMUSG00000023034                   &lt;NA&gt;  0.7549191 5.804234  6.649708
ENSMUSG00000032402                   &lt;NA&gt; -0.5181308 5.887314 -6.462453
ENSMUSG00000037868                   &lt;NA&gt;  0.9311988 0.373090  5.906601
ENSMUSG00000066687                   &lt;NA&gt; -0.7772013 4.422495 -5.816794
                        P.Value    adj.P.Val         B
ENSMUSG00000032846 1.348736e-09 2.968300e-05 11.819381
ENSMUSG00000053414 2.613976e-09 2.968300e-05 11.184022
ENSMUSG00000020423 6.956846e-09 5.266564e-05 10.266514
ENSMUSG00000021250 1.081866e-08 6.142565e-05  9.660660
ENSMUSG00000024222 1.630114e-08 7.404305e-05  9.321305
ENSMUSG00000037112 2.977792e-08 1.127144e-04  8.724483
ENSMUSG00000023034 5.671705e-08 1.840144e-04  7.926121
ENSMUSG00000032402 1.039351e-07 2.950588e-04  7.390251
ENSMUSG00000037868 6.302882e-07 1.590497e-03  5.856187
ENSMUSG00000066687 8.433688e-07 1.834372e-03  5.415240</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Interaction_Cortex"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000032846    chr13 107724617 107890064 165448      - HAVANA gene
ENSMUSG00000030324     chr6 115931748 115940036   8289      + HAVANA gene
ENSMUSG00000023978    chr17  46910459  46924933  14475      + HAVANA gene
ENSMUSG00000034837     chr9 107674474 107679712   5239      - HAVANA gene
ENSMUSG00000024215    chr17  27714352  27728955  14604      - HAVANA gene
ENSMUSG00000034057    chr10 116776535 116896919 120385      - HAVANA gene
ENSMUSG00000006007     chr1 150319417 150333906  14490      + HAVANA gene
ENSMUSG00000053414    chr16  90386013  90499553 113541      + HAVANA gene
ENSMUSG00000041534    chr14  33954003  33964216  10214      + HAVANA gene
ENSMUSG00000110715     chr8  97163793  97165030   1238      + HAVANA gene
                   bp_length phase            gene_id            gene_type
ENSMUSG00000032846      5550    NA ENSMUSG00000032846       protein_coding
ENSMUSG00000030324      6043    NA ENSMUSG00000030324       protein_coding
ENSMUSG00000023978      3002    NA ENSMUSG00000023978       protein_coding
ENSMUSG00000034837      3029    NA ENSMUSG00000034837       protein_coding
ENSMUSG00000024215      2305    NA ENSMUSG00000024215       protein_coding
ENSMUSG00000034057      3164    NA ENSMUSG00000034057       protein_coding
ENSMUSG00000006007      3350    NA ENSMUSG00000006007       protein_coding
ENSMUSG00000053414      8402    NA ENSMUSG00000053414       protein_coding
ENSMUSG00000041534      5295    NA ENSMUSG00000041534       protein_coding
ENSMUSG00000110715      1238    NA ENSMUSG00000110715 processed_pseudogene
                   gene_name level      mgi_id          havana_gene
ENSMUSG00000032846    Zswim6     2 MGI:1914513 OTTMUSG00000067880.2
ENSMUSG00000030324       Rho     2   MGI:97914 OTTMUSG00000056681.8
ENSMUSG00000023978     Prph2     2  MGI:102791 OTTMUSG00000034269.1
ENSMUSG00000034837     Gnat1     2   MGI:95778 OTTMUSG00000050785.4
ENSMUSG00000024215     Spdef     2 MGI:1353422 OTTMUSG00000021352.5
ENSMUSG00000034057     Myrfl     2 MGI:2685085 OTTMUSG00000064247.1
ENSMUSG00000006007       Pdc     2   MGI:98090 OTTMUSG00000049379.2
ENSMUSG00000053414      Hunk     2 MGI:1347352 OTTMUSG00000071848.1
ENSMUSG00000041534      Rbp3     2   MGI:97878 OTTMUSG00000069948.1
ENSMUSG00000110715    Gm5131     1 MGI:3644589 OTTMUSG00000061779.1
                                      tag      logFC   AveExpr         t
ENSMUSG00000032846 reference_genome_error  0.7844748  5.183653  6.298999
ENSMUSG00000030324                   &lt;NA&gt; -3.5928850  5.481363 -5.167467
ENSMUSG00000023978                   &lt;NA&gt; -4.5135240  2.513614 -4.267469
ENSMUSG00000034837                   &lt;NA&gt; -3.1894489  4.194471 -4.232893
ENSMUSG00000024215                   &lt;NA&gt; -6.4761811 -3.135970 -4.053863
ENSMUSG00000034057                   &lt;NA&gt; -5.9753354 -2.218449 -4.026197
ENSMUSG00000006007      overlapping_locus -6.3910979  1.198001 -4.002109
ENSMUSG00000053414                   &lt;NA&gt; -0.6236834  5.222101 -3.989510
ENSMUSG00000041534                   &lt;NA&gt; -3.6013727  3.290316 -3.981550
ENSMUSG00000110715         pseudo_consens  3.0086696 -1.376596  3.878041
                        P.Value   adj.P.Val           B
ENSMUSG00000032846 1.765067e-07 0.004008644  5.53664465
ENSMUSG00000030324 6.863485e-06 0.077938303  0.83324797
ENSMUSG00000023978 1.171170e-04 0.705940406 -3.46936415
ENSMUSG00000034837 1.302503e-04 0.705940406 -1.75020417
ENSMUSG00000024215 2.248674e-04 0.705940406 -4.41380392
ENSMUSG00000034057 2.445016e-04 0.705940406 -4.41760487
ENSMUSG00000006007 2.629458e-04 0.705940406 -4.15855843
ENSMUSG00000053414 2.731249e-04 0.705940406 -0.01891616
ENSMUSG00000041534 2.797527e-04 0.705940406 -3.23078823
ENSMUSG00000110715 3.815224e-04 0.768800381 -4.06748703</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Male_vs_female_Hippocampus"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000035150     chrX  94188707  94212862  24156      - HAVANA gene
ENSMUSG00000082424     chr2  11409031  11410030   1000      - HAVANA gene
ENSMUSG00000056673     chrY    897788    956786  58999      + HAVANA gene
ENSMUSG00000107951     chr6  75296118  75298059   1942      - HAVANA gene
ENSMUSG00000069045     chrY   1260771   1286629  25859      - HAVANA gene
ENSMUSG00000037369     chrX  18162575  18279936 117362      + HAVANA gene
ENSMUSG00000086503     chrX 103460366 103483254  22889      - HAVANA gene
ENSMUSG00000068457     chrY   1096861   1245759 148899      - HAVANA gene
ENSMUSG00000081214     chr1 181241988 181242320    333      - HAVANA gene
ENSMUSG00000046440     chr5 138504813 138505779    967      + HAVANA gene
                   bp_length phase            gene_id            gene_type
ENSMUSG00000035150      3812    NA ENSMUSG00000035150       protein_coding
ENSMUSG00000082424      1000    NA ENSMUSG00000082424 processed_pseudogene
ENSMUSG00000056673     15164    NA ENSMUSG00000056673       protein_coding
ENSMUSG00000107951      1942    NA ENSMUSG00000107951 processed_pseudogene
ENSMUSG00000069045      5199    NA ENSMUSG00000069045       protein_coding
ENSMUSG00000037369      8383    NA ENSMUSG00000037369       protein_coding
ENSMUSG00000086503     17946    NA ENSMUSG00000086503               lncRNA
ENSMUSG00000068457      9791    NA ENSMUSG00000068457       protein_coding
ENSMUSG00000081214       333    NA ENSMUSG00000081214 processed_pseudogene
ENSMUSG00000046440       967    NA ENSMUSG00000046440 processed_pseudogene
                    gene_name level      mgi_id          havana_gene
ENSMUSG00000035150    Eif2s3x     2 MGI:1349431 OTTMUSG00000018084.1
ENSMUSG00000082424    Gm13292     1 MGI:3649470 OTTMUSG00000011310.2
ENSMUSG00000056673      Kdm5d     2   MGI:99780 OTTMUSG00000045277.1
ENSMUSG00000107951     Gm6210     1 MGI:3643375 OTTMUSG00000057337.1
ENSMUSG00000069045      Ddx3y     2 MGI:1349406 OTTMUSG00000045279.1
ENSMUSG00000037369      Kdm6a     2 MGI:1095419 OTTMUSG00000016960.4
ENSMUSG00000086503       Xist     2   MGI:98974 OTTMUSG00000018295.5
ENSMUSG00000068457        Uty     2  MGI:894810 OTTMUSG00000032814.5
ENSMUSG00000081214 Rpl35a-ps2     2 MGI:3705461 OTTMUSG00000021804.1
ENSMUSG00000046440     Gm5564     1 MGI:3645018 OTTMUSG00000054712.1
                                 tag      logFC    AveExpr          t
ENSMUSG00000035150              &lt;NA&gt; -0.7030630  6.2281911 -10.710190
ENSMUSG00000082424    pseudo_consens  2.0344815  5.4398844   9.965207
ENSMUSG00000056673              &lt;NA&gt;  9.3988339 -0.4319924   9.874355
ENSMUSG00000107951    pseudo_consens -1.6268622  3.7621985  -9.475218
ENSMUSG00000069045 overlapping_locus 10.7644911  0.2849525   9.098790
ENSMUSG00000037369              &lt;NA&gt; -0.7093096  5.2968356  -8.791366
ENSMUSG00000086503        ncRNA_host -9.3537866  3.6571905  -8.653350
ENSMUSG00000068457              &lt;NA&gt;  9.9216952 -0.7664206   8.454778
ENSMUSG00000081214              &lt;NA&gt;  1.5005051  4.2444628   8.246348
ENSMUSG00000046440    pseudo_consens -0.8679116  4.6327367  -8.051157
                        P.Value    adj.P.Val         B
ENSMUSG00000035150 2.440610e-13 5.542870e-09 20.057730
ENSMUSG00000082424 2.038125e-12 2.008560e-08 18.142220
ENSMUSG00000056673 2.653199e-12 2.008560e-08  9.606961
ENSMUSG00000107951 8.558240e-12 4.859155e-08 16.764715
ENSMUSG00000069045 2.630388e-11 1.194775e-07  8.362526
ENSMUSG00000037369 6.665449e-11 2.522983e-07 14.495850
ENSMUSG00000086503 1.015537e-10 3.294838e-07 13.983020
ENSMUSG00000068457 1.868368e-10 5.304062e-07  6.042612
ENSMUSG00000081214 3.559771e-10 8.982885e-07 13.100793
ENSMUSG00000046440 6.537731e-10 1.484784e-06 12.214033</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Tamoxifen_effet_male_Hippocampus"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000022602    chr15  74669083  74672570   3488      - HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000025324     chr7  58656166  58829420 173255      + HAVANA gene
ENSMUSG00000019831    chr10  40883475  40938570  55096      + HAVANA gene
ENSMUSG00000022490    chr15 103530279 103538003   7725      - HAVANA gene
ENSMUSG00000068696     chr3 116249654 116253503   3850      - HAVANA gene
ENSMUSG00000071856    chr18  44425060  44812182 387123      - HAVANA gene
ENSMUSG00000046160    chr16  91269772  91271933   2162      + HAVANA gene
ENSMUSG00000071637    chr16  15887286  15891031   3746      + HAVANA gene
ENSMUSG00000055148     chr8  72319033  72321656   2624      + HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000022602      3057    NA ENSMUSG00000022602 protein_coding       Arc
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000025324      5587    NA ENSMUSG00000025324 protein_coding    Atp10a
ENSMUSG00000019831      2854    NA ENSMUSG00000019831 protein_coding     Wasf1
ENSMUSG00000022490      1306    NA ENSMUSG00000022490 protein_coding   Ppp1r1a
ENSMUSG00000068696      3504    NA ENSMUSG00000068696 protein_coding     Gpr88
ENSMUSG00000071856     14577    NA ENSMUSG00000071856 protein_coding       Mcc
ENSMUSG00000046160      2162    NA ENSMUSG00000046160 protein_coding     Olig1
ENSMUSG00000071637      3746    NA ENSMUSG00000071637 protein_coding     Cebpd
ENSMUSG00000055148      1847    NA ENSMUSG00000055148 protein_coding      Klf2
                   level      mgi_id          havana_gene  tag      logFC
ENSMUSG00000022602     2   MGI:88067 OTTMUSG00000016386.3 &lt;NA&gt;  0.5722024
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1 &lt;NA&gt; -0.6751847
ENSMUSG00000025324     2 MGI:1330809 OTTMUSG00000058914.1 &lt;NA&gt; -0.6831968
ENSMUSG00000019831     2 MGI:1890563 OTTMUSG00000020871.2 &lt;NA&gt;  0.2619162
ENSMUSG00000022490     2 MGI:1889595 OTTMUSG00000070242.1 &lt;NA&gt;  0.3924606
ENSMUSG00000068696     2 MGI:1927653 OTTMUSG00000054451.3 &lt;NA&gt; -2.0153914
ENSMUSG00000071856     2   MGI:96930 OTTMUSG00000055565.1 &lt;NA&gt; -0.4282978
ENSMUSG00000046160     2 MGI:1355334 OTTMUSG00000019872.1 &lt;NA&gt;  0.4698001
ENSMUSG00000071637     2  MGI:103573 OTTMUSG00000060220.1 &lt;NA&gt;  0.6781388
ENSMUSG00000055148     2 MGI:1342772 OTTMUSG00000022542.1 &lt;NA&gt;  0.5403129
                    AveExpr         t      P.Value    adj.P.Val        B
ENSMUSG00000022602 5.636934  7.139420 1.172851e-08 0.0002663662 9.798037
ENSMUSG00000066687 4.422495 -6.758041 3.997735e-08 0.0004539628 8.616704
ENSMUSG00000025324 2.959977 -6.136552 2.989251e-07 0.0022629624 6.507270
ENSMUSG00000019831 6.899311  6.002020 4.624995e-07 0.0026259568 6.219988
ENSMUSG00000022490 5.576759  5.300137 4.480983e-06 0.0198694827 4.114949
ENSMUSG00000068696 2.938554 -5.191250 6.359103e-06 0.0198694827 3.069557
ENSMUSG00000071856 4.620932 -5.177194 6.652567e-06 0.0198694827 3.776817
ENSMUSG00000046160 2.038781  5.161370 6.999069e-06 0.0198694827 3.677239
ENSMUSG00000071637 1.840047  4.917549 1.525677e-05 0.0384996066 2.953611
ENSMUSG00000055148 2.371853  4.827864 2.028729e-05 0.0407714245 2.691129</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Tamoxifen_effet_female_Hippocampus"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000025316     chr8 121949750 122029258  79509      + HAVANA gene
ENSMUSG00000004328     chr7  17031507  17062427  30921      - HAVANA gene
ENSMUSG00000054720     chr5 105519388 105613018  93631      + HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000050919    chr13  99184823  99250656  65834      + HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000048701    chr10  70097121  70193200  96080      + HAVANA gene
ENSMUSG00000025931     chr1  20890606  20939650  49045      + HAVANA gene
ENSMUSG00000035877     chr2 160748708 160872998 124291      - HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000025316      6632    NA ENSMUSG00000025316 protein_coding      Banp
ENSMUSG00000004328      6710    NA ENSMUSG00000004328 protein_coding     Hif3a
ENSMUSG00000054720      7654    NA ENSMUSG00000054720 protein_coding    Lrrc8c
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000050919      6382    NA ENSMUSG00000050919 protein_coding    Zfp366
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000048701      6385    NA ENSMUSG00000048701 protein_coding     Ccdc6
ENSMUSG00000025931      5488    NA ENSMUSG00000025931 protein_coding     Paqr8
ENSMUSG00000035877     13259    NA ENSMUSG00000035877 protein_coding      Zhx3
                   level      mgi_id          havana_gene               tag
ENSMUSG00000025316     1 MGI:1889023 OTTMUSG00000036899.8 overlapping_locus
ENSMUSG00000004328     2 MGI:1859778 OTTMUSG00000022817.2              &lt;NA&gt;
ENSMUSG00000054720     2 MGI:2140839 OTTMUSG00000026735.2 overlapping_locus
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
ENSMUSG00000050919     2 MGI:2178429 OTTMUSG00000027698.2              &lt;NA&gt;
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3              &lt;NA&gt;
ENSMUSG00000048701     2 MGI:1923801 OTTMUSG00000032767.1              &lt;NA&gt;
ENSMUSG00000025931     2 MGI:1921479 OTTMUSG00000048364.1              &lt;NA&gt;
ENSMUSG00000035877     2 MGI:2444772 OTTMUSG00000001048.8 overlapping_locus
                        logFC  AveExpr         t      P.Value    adj.P.Val
ENSMUSG00000025316 -0.8061109 4.583836 -8.298096 3.031959e-10 6.885882e-06
ENSMUSG00000004328 -1.3895679 2.342140 -7.461895 4.190355e-09 4.758358e-05
ENSMUSG00000054720 -0.5270337 4.741372 -7.078938 1.423781e-08 1.077850e-04
ENSMUSG00000021250  0.7716723 4.877409  6.321391 1.641488e-07 9.319957e-04
ENSMUSG00000050919 -0.6913393 2.628260 -5.779461 9.518678e-07 4.323574e-03
ENSMUSG00000066687 -0.5915626 4.422495 -5.458682 2.687678e-06 1.017331e-02
ENSMUSG00000037868  1.1601114 0.373090  5.300993 4.468656e-06 1.449824e-02
ENSMUSG00000048701 -0.3087373 5.650159 -5.207040 6.044720e-06 1.716020e-02
ENSMUSG00000025931 -0.2926002 6.125810 -5.037738 1.039850e-05 2.544713e-02
ENSMUSG00000035877 -0.3270911 6.115502 -5.014364 1.120476e-05 2.544713e-02
                           B
ENSMUSG00000025316 13.199879
ENSMUSG00000004328  9.935354
ENSMUSG00000054720  9.609580
ENSMUSG00000021250  7.252956
ENSMUSG00000050919  5.557808
ENSMUSG00000066687  4.571148
ENSMUSG00000037868  3.868690
ENSMUSG00000048701  3.778299
ENSMUSG00000025931  3.191117
ENSMUSG00000035877  3.168384</code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Interaction_Hippocampus"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand  source type
ENSMUSG00000093185     chr4  45396994  45397118    125      - ENSEMBL gene
ENSMUSG00000004328     chr7  17031507  17062427  30921      -  HAVANA gene
ENSMUSG00000037754     chr2 158665398 158766334 100937      +  HAVANA gene
ENSMUSG00000067575    chr17  29136972  29137304    333      -  HAVANA gene
ENSMUSG00000106895     chr5  41342123  41342931    809      +  HAVANA gene
ENSMUSG00000026496     chr1 180568924 180601254  32331      +  HAVANA gene
ENSMUSG00000082424     chr2  11409031  11410030   1000      -  HAVANA gene
ENSMUSG00000082029     chr2 119665626 119666036    411      -  HAVANA gene
ENSMUSG00000019831    chr10  40883475  40938570  55096      +  HAVANA gene
ENSMUSG00000054720     chr5 105519388 105613018  93631      +  HAVANA gene
                   bp_length phase            gene_id            gene_type
ENSMUSG00000093185       125    NA ENSMUSG00000093185                miRNA
ENSMUSG00000004328      6710    NA ENSMUSG00000004328       protein_coding
ENSMUSG00000037754      8844    NA ENSMUSG00000037754       protein_coding
ENSMUSG00000067575       333    NA ENSMUSG00000067575 processed_pseudogene
ENSMUSG00000106895       809    NA ENSMUSG00000106895 processed_pseudogene
ENSMUSG00000026496      5293    NA ENSMUSG00000026496       protein_coding
ENSMUSG00000082424      1000    NA ENSMUSG00000082424 processed_pseudogene
ENSMUSG00000082029       411    NA ENSMUSG00000082029 processed_pseudogene
ENSMUSG00000019831      2854    NA ENSMUSG00000019831       protein_coding
ENSMUSG00000054720      7654    NA ENSMUSG00000054720       protein_coding
                    gene_name level      mgi_id          havana_gene
ENSMUSG00000093185    Gm25692     3 MGI:5455469                 &lt;NA&gt;
ENSMUSG00000004328      Hif3a     2 MGI:1859778 OTTMUSG00000022817.2
ENSMUSG00000037754   Ppp1r16b     2 MGI:2151841 OTTMUSG00000015931.2
ENSMUSG00000067575 Rpl35a-ps3     2 MGI:3704473 OTTMUSG00000031659.1
ENSMUSG00000106895     Gm4754     1 MGI:3645925 OTTMUSG00000055789.1
ENSMUSG00000026496      Parp1     2 MGI:1340806 OTTMUSG00000050450.2
ENSMUSG00000082424    Gm13292     1 MGI:3649470 OTTMUSG00000011310.2
ENSMUSG00000082029      H3f3c     2 MGI:3650546 OTTMUSG00000016509.1
ENSMUSG00000019831      Wasf1     2 MGI:1890563 OTTMUSG00000020871.2
ENSMUSG00000054720     Lrrc8c     2 MGI:2140839 OTTMUSG00000026735.2
                                 tag      logFC   AveExpr         t
ENSMUSG00000093185              &lt;NA&gt;  0.9495883 0.9472638  5.038308
ENSMUSG00000004328              &lt;NA&gt;  1.2763016 2.3421403  5.028972
ENSMUSG00000037754        ncRNA_host  0.4188683 6.1623962  4.615291
ENSMUSG00000067575              &lt;NA&gt; -3.2179618 0.9348731 -4.609530
ENSMUSG00000106895    pseudo_consens -2.0513510 3.0253387 -4.564597
ENSMUSG00000026496        ncRNA_host  0.2895852 6.4079653  4.370886
ENSMUSG00000082424    pseudo_consens -1.1367397 5.4398844 -4.343326
ENSMUSG00000082029              &lt;NA&gt; -0.5993018 4.1041019 -4.228990
ENSMUSG00000019831              &lt;NA&gt;  0.2687815 6.8993107  4.207258
ENSMUSG00000054720 overlapping_locus  0.4284844 4.7413717  4.178376
                        P.Value adj.P.Val         B
ENSMUSG00000093185 1.037956e-05 0.1214346 3.2640809
ENSMUSG00000004328 1.069390e-05 0.1214346 3.3126177
ENSMUSG00000037754 3.969062e-05 0.2113611 1.7803189
ENSMUSG00000067575 4.041525e-05 0.2113611 0.4404005
ENSMUSG00000106895 4.653277e-05 0.2113611 1.6669993
ENSMUSG00000026496 8.509859e-05 0.3006862 1.0642724
ENSMUSG00000082424 9.267770e-05 0.3006862 1.2326609
ENSMUSG00000082029 1.318200e-04 0.3495943 0.9309987
ENSMUSG00000019831 1.409057e-04 0.3495943 0.4690830
ENSMUSG00000054720 1.539317e-04 0.3495943 0.5999630</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Male_vs_female_Retina"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand  source type
ENSMUSG00000037369     chrX  18162575  18279936 117362      +  HAVANA gene
ENSMUSG00000098262     chrX 103480946 103481026     81      + ENSEMBL gene
ENSMUSG00000068457     chrY   1096861   1245759 148899      -  HAVANA gene
ENSMUSG00000035150     chrX  94188707  94212862  24156      -  HAVANA gene
ENSMUSG00000099312     chrX 103461218 103461285     68      - ENSEMBL gene
ENSMUSG00000056673     chrY    897788    956786  58999      +  HAVANA gene
ENSMUSG00000086503     chrX 103460366 103483254  22889      -  HAVANA gene
ENSMUSG00000032599     chr9 108783796 108806337  22542      +  HAVANA gene
ENSMUSG00000025332     chrX 152233020 152274535  41516      +  HAVANA gene
ENSMUSG00000069045     chrY   1260771   1286629  25859      -  HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000037369      8383    NA ENSMUSG00000037369 protein_coding     Kdm6a
ENSMUSG00000098262        81    NA ENSMUSG00000098262       misc_RNA   Gm27520
ENSMUSG00000068457      9791    NA ENSMUSG00000068457 protein_coding       Uty
ENSMUSG00000035150      3812    NA ENSMUSG00000035150 protein_coding   Eif2s3x
ENSMUSG00000099312        68    NA ENSMUSG00000099312       misc_RNA   Gm27733
ENSMUSG00000056673     15164    NA ENSMUSG00000056673 protein_coding     Kdm5d
ENSMUSG00000086503     17946    NA ENSMUSG00000086503         lncRNA      Xist
ENSMUSG00000032599      4067    NA ENSMUSG00000032599 protein_coding     Ip6k2
ENSMUSG00000025332      7110    NA ENSMUSG00000025332 protein_coding     Kdm5c
ENSMUSG00000069045      5199    NA ENSMUSG00000069045 protein_coding     Ddx3y
                   level      mgi_id          havana_gene               tag
ENSMUSG00000037369     2 MGI:1095419 OTTMUSG00000016960.4              &lt;NA&gt;
ENSMUSG00000098262     3 MGI:5530902                 &lt;NA&gt;              &lt;NA&gt;
ENSMUSG00000068457     2  MGI:894810 OTTMUSG00000032814.5              &lt;NA&gt;
ENSMUSG00000035150     2 MGI:1349431 OTTMUSG00000018084.1              &lt;NA&gt;
ENSMUSG00000099312     3 MGI:5531115                 &lt;NA&gt;              &lt;NA&gt;
ENSMUSG00000056673     2   MGI:99780 OTTMUSG00000045277.1              &lt;NA&gt;
ENSMUSG00000086503     2   MGI:98974 OTTMUSG00000018295.5        ncRNA_host
ENSMUSG00000032599     2 MGI:1923750 OTTMUSG00000050674.3              &lt;NA&gt;
ENSMUSG00000025332     2   MGI:99781 OTTMUSG00000017219.2              &lt;NA&gt;
ENSMUSG00000069045     2 MGI:1349406 OTTMUSG00000045279.1 overlapping_locus
                         logFC    AveExpr          t      P.Value    adj.P.Val
ENSMUSG00000037369  -0.8243572  5.2968356 -10.256522 8.812143e-13 2.001326e-08
ENSMUSG00000098262  -8.4438064 -3.2312004  -9.348773 1.245473e-11 1.179526e-07
ENSMUSG00000068457  11.1463406 -0.7664206   9.273620 1.558090e-11 1.179526e-07
ENSMUSG00000035150  -0.6300972  6.2281911  -9.140256 2.322368e-11 1.318583e-07
ENSMUSG00000099312  -8.7868844 -2.8949791  -8.966557 3.918392e-11 1.779812e-07
ENSMUSG00000056673  10.1313525 -0.4319924   8.787482 6.744686e-11 2.552976e-07
ENSMUSG00000086503 -11.3063140  3.6571905  -8.696496 8.900726e-11 2.887777e-07
ENSMUSG00000032599  -0.7804967  6.8000860  -8.581005 1.267453e-10 3.598140e-07
ENSMUSG00000025332  -0.5385336  6.4390603  -8.365994 2.457333e-10 6.200944e-07
ENSMUSG00000069045  10.6453161  0.2849525   7.838681 1.272686e-09 2.890398e-06
                           B
ENSMUSG00000037369 18.755723
ENSMUSG00000098262  9.483437
ENSMUSG00000068457  8.006425
ENSMUSG00000035150 15.421976
ENSMUSG00000099312  8.939016
ENSMUSG00000056673  7.371301
ENSMUSG00000086503 13.967114
ENSMUSG00000032599 13.604707
ENSMUSG00000025332 12.964286
ENSMUSG00000069045  6.303909</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Tamoxifen_effet_male_Retina"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000038418    chr18  34859823  34864984   5162      + HAVANA gene
ENSMUSG00000021091    chr12 104406729 104414329   7601      + HAVANA gene
ENSMUSG00000028777     chr5  17962549  18019834  57286      + HAVANA gene
ENSMUSG00000022382    chr15  85535437  85582473  47037      - HAVANA gene
ENSMUSG00000050440     chr7  30942368  30944032   1665      - HAVANA gene
ENSMUSG00000021177    chr12  99884517  99955219  70703      + HAVANA gene
ENSMUSG00000090113    chr17  25942233  25944961   2729      - HAVANA gene
ENSMUSG00000000628     chr6  82725025  82774454  49430      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000038418      4484    NA ENSMUSG00000038418 protein_coding      Egr1
ENSMUSG00000021091      2064    NA ENSMUSG00000021091 protein_coding Serpina3n
ENSMUSG00000028777      1361    NA ENSMUSG00000028777 protein_coding     Gnat3
ENSMUSG00000022382      4898    NA ENSMUSG00000022382 protein_coding     Wnt7b
ENSMUSG00000050440       410    NA ENSMUSG00000050440 protein_coding      Hamp
ENSMUSG00000021177      7988    NA ENSMUSG00000021177 protein_coding      Tdp1
ENSMUSG00000090113      2318    NA ENSMUSG00000090113 protein_coding    Nhlrc4
ENSMUSG00000000628      5969    NA ENSMUSG00000000628 protein_coding       Hk2
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
                   level      mgi_id          havana_gene               tag
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000038418     2   MGI:95295 OTTMUSG00000021064.2              &lt;NA&gt;
ENSMUSG00000021091     2  MGI:105045 OTTMUSG00000016315.1              &lt;NA&gt;
ENSMUSG00000028777     2 MGI:3588268 OTTMUSG00000053053.1              &lt;NA&gt;
ENSMUSG00000022382     2   MGI:98962 OTTMUSG00000070694.1              &lt;NA&gt;
ENSMUSG00000050440     2 MGI:1933533 OTTMUSG00000058034.2              &lt;NA&gt;
ENSMUSG00000021177     2 MGI:1920036 OTTMUSG00000033586.3              &lt;NA&gt;
ENSMUSG00000090113     2 MGI:3687200 OTTMUSG00000033452.3 overlapping_locus
ENSMUSG00000000628     1 MGI:1315197 OTTMUSG00000035991.3              &lt;NA&gt;
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
                        logFC   AveExpr         t      P.Value  adj.P.Val
ENSMUSG00000066687 -1.4713200  4.422495 -5.291732 4.603863e-06 0.09091648
ENSMUSG00000038418  0.9282016  6.923094  5.114657 8.129704e-06 0.09091648
ENSMUSG00000021091 -1.3872732  4.291497 -4.992636 1.200957e-05 0.09091648
ENSMUSG00000028777 -3.0406224 -5.300356 -4.667962 3.363063e-05 0.19094632
ENSMUSG00000022382  5.8028790  2.419470  4.482415 6.016417e-05 0.26773672
ENSMUSG00000050440 -2.3102764 -5.182209 -4.430451 7.073314e-05 0.26773672
ENSMUSG00000021177 -0.3984100  3.639913 -4.185343 1.506859e-04 0.48888972
ENSMUSG00000090113 -1.7877501 -1.400430 -4.088057 2.027133e-04 0.49219913
ENSMUSG00000000628 -0.2711272  4.338623 -4.077455 2.093443e-04 0.49219913
ENSMUSG00000021250  0.5089485  4.877409  4.050158 2.274050e-04 0.49219913
                            B
ENSMUSG00000066687  2.3574621
ENSMUSG00000038418  3.4685471
ENSMUSG00000021091  3.0160388
ENSMUSG00000028777 -3.7983715
ENSMUSG00000022382 -3.8816016
ENSMUSG00000050440 -3.6209524
ENSMUSG00000021177  0.7288394
ENSMUSG00000090113 -3.0383632
ENSMUSG00000000628  0.5245150
ENSMUSG00000021250  0.5349818</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Tamoxifen_effet_female_Retina"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000113671    chr13  43948879  43952699   3821      + HAVANA gene
ENSMUSG00000007682    chr12  90724552  90738438  13887      - HAVANA gene
ENSMUSG00000033618    chr16  21794346  21933439 139094      + HAVANA gene
ENSMUSG00000036390     chr6  67035096  67037457   2362      - HAVANA gene
ENSMUSG00000114414    chr13  99007099  99021695  14597      + HAVANA gene
ENSMUSG00000054720     chr5 105519388 105613018  93631      + HAVANA gene
ENSMUSG00000026885     chr2  35751241  35979913 228673      - HAVANA gene
ENSMUSG00000026773     chr2  11471433  11554077  82645      - HAVANA gene
ENSMUSG00000000628     chr6  82725025  82774454  49430      - HAVANA gene
ENSMUSG00000112264    chr10  96495721  96523982  28262      - HAVANA gene
                   bp_length phase            gene_id      gene_type
ENSMUSG00000113671      1916    NA ENSMUSG00000113671         lncRNA
ENSMUSG00000007682      5815    NA ENSMUSG00000007682 protein_coding
ENSMUSG00000033618      8714    NA ENSMUSG00000033618 protein_coding
ENSMUSG00000036390      1764    NA ENSMUSG00000036390 protein_coding
ENSMUSG00000114414       974    NA ENSMUSG00000114414         lncRNA
ENSMUSG00000054720      7654    NA ENSMUSG00000054720 protein_coding
ENSMUSG00000026885      4024    NA ENSMUSG00000026885 protein_coding
ENSMUSG00000026773      7561    NA ENSMUSG00000026773 protein_coding
ENSMUSG00000000628      5969    NA ENSMUSG00000000628 protein_coding
ENSMUSG00000112264      2031    NA ENSMUSG00000112264         lncRNA
                       gene_name level      mgi_id          havana_gene
ENSMUSG00000113671        Gm2233     2 MGI:3780403 OTTMUSG00000067460.1
ENSMUSG00000007682          Dio2     2 MGI:1338833 OTTMUSG00000065914.1
ENSMUSG00000033618       Map3k13     2 MGI:2444243 OTTMUSG00000071512.1
ENSMUSG00000036390       Gadd45a     2  MGI:107799 OTTMUSG00000023157.2
ENSMUSG00000114414 A930014D07Rik     2 MGI:1925197 OTTMUSG00000067863.1
ENSMUSG00000054720        Lrrc8c     2 MGI:2140839 OTTMUSG00000026735.2
ENSMUSG00000026885        Ttll11     2 MGI:1921660 OTTMUSG00000012151.4
ENSMUSG00000026773        Pfkfb3     2 MGI:2181202 OTTMUSG00000011314.6
ENSMUSG00000000628           Hk2     1 MGI:1315197 OTTMUSG00000035991.3
ENSMUSG00000112264       Gm48505     2 MGI:6098035 OTTMUSG00000064509.1
                                 tag      logFC   AveExpr         t
ENSMUSG00000113671              &lt;NA&gt;  1.0326769 -3.858227  6.498729
ENSMUSG00000007682              &lt;NA&gt; -0.8734614  6.126186 -6.160339
ENSMUSG00000033618 overlapping_locus -0.4609769  5.487564 -6.068774
ENSMUSG00000036390              &lt;NA&gt;  0.5148577  2.993659  5.490786
ENSMUSG00000114414              &lt;NA&gt;  1.2080974 -4.112565  5.487216
ENSMUSG00000054720 overlapping_locus -0.6873177  4.741372 -5.423544
ENSMUSG00000026885 overlapping_locus  0.6784458  5.064492  5.324012
ENSMUSG00000026773 overlapping_locus -0.4967439  5.475579 -5.254739
ENSMUSG00000000628              &lt;NA&gt; -0.3677290  4.338623 -5.096236
ENSMUSG00000112264 overlapping_locus  0.5420246 -2.947248  5.037605
                        P.Value   adj.P.Val        B
ENSMUSG00000113671 9.241905e-08 0.002098929 7.713639
ENSMUSG00000007682 2.767263e-07 0.002819496 6.455247
ENSMUSG00000033618 3.724401e-07 0.002819496 6.075613
ENSMUSG00000036390 2.422966e-06 0.011133255 4.474606
ENSMUSG00000114414 2.451071e-06 0.011133255 4.552475
ENSMUSG00000054720 3.010477e-06 0.011395158 4.345214
ENSMUSG00000026885 4.149442e-06 0.013462568 3.843683
ENSMUSG00000026773 5.185657e-06 0.014721431 3.518262
ENSMUSG00000000628 8.623791e-06 0.021761658 2.737529
ENSMUSG00000112264 1.040290e-05 0.023626034 3.152603</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef=</span><span class="st">"Interaction_Retina"</span>, <span class="at">n=</span><span class="dv">10</span>, <span class="at">sort.by=</span><span class="st">"P"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end width strand source type
ENSMUSG00000050440     chr7  30942368  30944032  1665      - HAVANA gene
ENSMUSG00000113671    chr13  43948879  43952699  3821      + HAVANA gene
ENSMUSG00000032599     chr9 108783796 108806337 22542      + HAVANA gene
ENSMUSG00000082415     chr3 101206859 101207399   541      - HAVANA gene
ENSMUSG00000108033     chr6  42328713  42350862 22150      - HAVANA gene
ENSMUSG00000028777     chr5  17962549  18019834 57286      + HAVANA gene
ENSMUSG00000004371     chr7   4772373   4782859 10487      - HAVANA gene
ENSMUSG00000085900     chr5 107630250 107633850  3601      - HAVANA gene
ENSMUSG00000091462    chr13  34034526  34035036   511      + HAVANA gene
ENSMUSG00000032010     chr9  44067021  44095627 28607      + HAVANA gene
                   bp_length phase            gene_id            gene_type
ENSMUSG00000050440       410    NA ENSMUSG00000050440       protein_coding
ENSMUSG00000113671      1916    NA ENSMUSG00000113671               lncRNA
ENSMUSG00000032599      4067    NA ENSMUSG00000032599       protein_coding
ENSMUSG00000082415       541    NA ENSMUSG00000082415 processed_pseudogene
ENSMUSG00000108033      3444    NA ENSMUSG00000108033               lncRNA
ENSMUSG00000028777      1361    NA ENSMUSG00000028777       protein_coding
ENSMUSG00000004371      2081    NA ENSMUSG00000004371       protein_coding
ENSMUSG00000085900       537    NA ENSMUSG00000085900               lncRNA
ENSMUSG00000091462       511    NA ENSMUSG00000091462 processed_pseudogene
ENSMUSG00000032010      4818    NA ENSMUSG00000032010       protein_coding
                       gene_name level      mgi_id          havana_gene
ENSMUSG00000050440          Hamp     2 MGI:1933533 OTTMUSG00000058034.2
ENSMUSG00000113671        Gm2233     2 MGI:3780403 OTTMUSG00000067460.1
ENSMUSG00000032599         Ip6k2     2 MGI:1923750 OTTMUSG00000050674.3
ENSMUSG00000082415       Gm12486     1 MGI:3650949 OTTMUSG00000007091.1
ENSMUSG00000108033       Gm44284     2 MGI:5690676 OTTMUSG00000057545.1
ENSMUSG00000028777         Gnat3     2 MGI:3588268 OTTMUSG00000053053.1
ENSMUSG00000004371          Il11     2  MGI:107613 OTTMUSG00000040054.4
ENSMUSG00000085900 A930041C12Rik     2 MGI:1925693 OTTMUSG00000023397.1
ENSMUSG00000091462       Gm17084     2 MGI:4937911 OTTMUSG00000035492.3
ENSMUSG00000032010          Usp2     2 MGI:1858178 OTTMUSG00000042086.3
                              tag      logFC    AveExpr         t      P.Value
ENSMUSG00000050440           &lt;NA&gt; -3.9837473 -5.1822093 -5.515318 2.238313e-06
ENSMUSG00000113671           &lt;NA&gt; -1.0240985 -3.8582273 -4.844925 1.921823e-05
ENSMUSG00000032599           &lt;NA&gt;  0.5664970  6.8000860  4.566019 4.632594e-05
ENSMUSG00000082415 pseudo_consens -3.5233333 -4.8834192 -4.426285 7.165533e-05
ENSMUSG00000108033     ncRNA_host -5.4271229 -1.3458525 -4.249454 1.237893e-04
ENSMUSG00000028777           &lt;NA&gt; -3.5078529 -5.3003561 -4.243157 1.262081e-04
ENSMUSG00000004371           &lt;NA&gt;  5.6065798 -2.5096667  4.181637 1.524044e-04
ENSMUSG00000085900           &lt;NA&gt; -0.7428255 -2.5832630 -4.072116 2.127638e-04
ENSMUSG00000091462           &lt;NA&gt; -2.3530409 -0.6305491 -4.048959 2.282321e-04
ENSMUSG00000032010           &lt;NA&gt;  0.3645928  6.1176977  4.034685 2.383065e-04
                    adj.P.Val          B
ENSMUSG00000050440 0.05083434 -2.1722829
ENSMUSG00000113671 0.21823264  1.9051464
ENSMUSG00000032599 0.35070281  1.8162461
ENSMUSG00000082415 0.40684106 -2.8090487
ENSMUSG00000108033 0.47771864 -3.5192936
ENSMUSG00000028777 0.47771864 -3.1007992
ENSMUSG00000004371 0.49446517 -3.4791193
ENSMUSG00000085900 0.50159846  0.5644998
ENSMUSG00000091462 0.50159846 -1.6650232
ENSMUSG00000032010 0.50159846  0.3153511</code></pre>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> Do the number of DE genes correspond to your expectations from the PCA? What would you conclude on the difference of Tamoxifen effects in male and female?</p>
<p><img src="round-help-button.png" class="img-fluid"> What does this tissue-by-tissue analysis reveals that was not so apparent in the <code>hands_on_limma.qmd</code> script (where tissue variation was absorbed by the model)?</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-7_6f56bed44b57e7226bbc5b907154d7c8">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot expression of top genes: choose one contrast For example: sex</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="do">## differences in Cortex, top 10 genes</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Male_vs_female_Cortex"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm[top<span class="sc">$</span>gene_id, ] <span class="sc">|&gt;</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">col =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name, <span class="at">scales =</span> <span class="st">"free_y"</span>,</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><img src="wrench-and-hammer.png" class="img-fluid"> Explore the results from the different contrasts</p>
<p><img src="round-help-button.png" class="img-fluid"> What do you notice? Are some of these genes described in the original paper as showing tissue differences?</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-8_ef4ab1aa258a88cd3f1082f05c4436ee">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">## See Fig 4e and f</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Tamoxifen_effet_male_Cortex"</span>, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">n =</span> <span class="cn">Inf</span>,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>top[top<span class="sc">$</span>gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Egr2"</span>, <span class="st">"Fos"</span>, <span class="st">"Dusp1"</span>, <span class="st">"Nr4a1"</span>, <span class="st">"Sik1"</span>, <span class="st">"Arc"</span>, <span class="st">"Egr1"</span>,</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Plcl2"</span>, <span class="st">"Galnt9"</span>, <span class="st">"Per2"</span>, <span class="st">"Zbtb16"</span>, <span class="st">"Map3k13"</span>, <span class="st">"Banp"</span>), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000022602    chr15  74669083  74672570   3488      - HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000024190    chr17  26505590  26562128  56539      - HAVANA gene
ENSMUSG00000023034    chr15 101254269 101274795  20527      + HAVANA gene
ENSMUSG00000024042    chr17  31844250  31855804  11555      - HAVANA gene
ENSMUSG00000025316     chr8 121949750 122029258  79509      + HAVANA gene
ENSMUSG00000055866     chr1  91415982  91459324  43343      - HAVANA gene
ENSMUSG00000033316     chr5 110544355 110621380  77026      + HAVANA gene
ENSMUSG00000038418    chr18  34859823  34864984   5162      + HAVANA gene
ENSMUSG00000033618    chr16  21794346  21933439 139094      + HAVANA gene
ENSMUSG00000038910    chr17  50509403  50688484 179082      + HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000022602      3057    NA ENSMUSG00000022602 protein_coding       Arc
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000024190      3311    NA ENSMUSG00000024190 protein_coding     Dusp1
ENSMUSG00000023034      3509    NA ENSMUSG00000023034 protein_coding     Nr4a1
ENSMUSG00000024042      5133    NA ENSMUSG00000024042 protein_coding      Sik1
ENSMUSG00000025316      6632    NA ENSMUSG00000025316 protein_coding      Banp
ENSMUSG00000055866      5856    NA ENSMUSG00000055866 protein_coding      Per2
ENSMUSG00000033316      3006    NA ENSMUSG00000033316 protein_coding    Galnt9
ENSMUSG00000038418      4484    NA ENSMUSG00000038418 protein_coding      Egr1
ENSMUSG00000033618      8714    NA ENSMUSG00000033618 protein_coding   Map3k13
ENSMUSG00000038910      6626    NA ENSMUSG00000038910 protein_coding     Plcl2
                   level      mgi_id          havana_gene               tag
ENSMUSG00000022602     2   MGI:88067 OTTMUSG00000016386.3              &lt;NA&gt;
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3              &lt;NA&gt;
ENSMUSG00000024190     2  MGI:105120 OTTMUSG00000020083.4              &lt;NA&gt;
ENSMUSG00000023034     2 MGI:1352454 OTTMUSG00000071152.1              &lt;NA&gt;
ENSMUSG00000024042     2  MGI:104754 OTTMUSG00000073186.1              &lt;NA&gt;
ENSMUSG00000025316     1 MGI:1889023 OTTMUSG00000036899.8 overlapping_locus
ENSMUSG00000055866     2 MGI:1195265 OTTMUSG00000048493.1              &lt;NA&gt;
ENSMUSG00000033316     2 MGI:2677965 OTTMUSG00000054216.1              &lt;NA&gt;
ENSMUSG00000038418     2   MGI:95295 OTTMUSG00000021064.2              &lt;NA&gt;
ENSMUSG00000033618     2 MGI:2444243 OTTMUSG00000071512.1 overlapping_locus
ENSMUSG00000038910     2 MGI:1352756 OTTMUSG00000072532.1              &lt;NA&gt;
                        logFC  AveExpr         t      P.Value    adj.P.Val
ENSMUSG00000022602  0.5678203 5.636934  6.781845 3.702287e-08 0.0008408263
ENSMUSG00000066687 -0.8473846 4.422495 -6.406379 1.246328e-07 0.0014152681
ENSMUSG00000021250  0.6957341 4.877409  6.234880 2.172963e-07 0.0016450056
ENSMUSG00000037868  0.7913071 0.373090  4.732882 2.740327e-05 0.1037259411
ENSMUSG00000024190  0.4288882 4.473001  3.746019 5.643175e-04 0.8544143401
ENSMUSG00000023034  0.4146351 5.804234  3.627952 7.973726e-04 0.9347329889
ENSMUSG00000024042  0.5547683 3.747410  3.420807 1.446609e-03 0.9999849225
ENSMUSG00000025316 -0.3919869 4.583836 -3.367639 1.681552e-03 0.9999849225
ENSMUSG00000055866 -0.3936909 4.916803 -3.244630 2.372390e-03 0.9999849225
ENSMUSG00000033316 -0.2901871 5.498505 -2.879929 6.350262e-03 0.9999849225
ENSMUSG00000038418  0.3276679 6.923094  2.678329 1.066845e-02 0.9999849225
ENSMUSG00000033618 -0.2174570 5.487564 -2.549556 1.470604e-02 0.9999849225
ENSMUSG00000038910 -0.1477936 5.382672 -1.929956 6.069501e-02 0.9999849225
                            B
ENSMUSG00000022602  7.7699626
ENSMUSG00000066687  5.5696775
ENSMUSG00000021250  5.3990485
ENSMUSG00000037868  0.8368334
ENSMUSG00000024190 -0.5401766
ENSMUSG00000023034 -0.6371147
ENSMUSG00000024042 -1.5568783
ENSMUSG00000025316 -1.5036834
ENSMUSG00000055866 -1.6306593
ENSMUSG00000033316 -2.2969033
ENSMUSG00000038418 -2.7344861
ENSMUSG00000033618 -2.9608716
ENSMUSG00000038910 -4.0177044</code></pre>
</div>
</div>
<section id="interactions-term" class="level2">
<h2 class="anchored" data-anchor-id="interactions-term">Interactions term</h2>
<p>We will now look at the genes reacting differently to Tamoxifen treatment in male and female. The genes reaching signficiance in the different tissues are distinct</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-9_ec3d71679b9a01011e345a3cba4d7235">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Genes significant wiht a relaxed FDR cutoff of 20%</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Cortex"</span>, <span class="at">p.value =</span> <span class="fl">0.2</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000032846    chr13 107724617 107890064 165448      - HAVANA gene
ENSMUSG00000030324     chr6 115931748 115940036   8289      + HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000032846      5550    NA ENSMUSG00000032846 protein_coding    Zswim6
ENSMUSG00000030324      6043    NA ENSMUSG00000030324 protein_coding       Rho
                   level      mgi_id          havana_gene
ENSMUSG00000032846     2 MGI:1914513 OTTMUSG00000067880.2
ENSMUSG00000030324     2   MGI:97914 OTTMUSG00000056681.8
                                      tag      logFC  AveExpr         t
ENSMUSG00000032846 reference_genome_error  0.7844748 5.183653  6.298999
ENSMUSG00000030324                   &lt;NA&gt; -3.5928850 5.481363 -5.167467
                        P.Value   adj.P.Val        B
ENSMUSG00000032846 1.765067e-07 0.004008644 5.536645
ENSMUSG00000030324 6.863485e-06 0.077938303 0.833248</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Hippocampus"</span>, <span class="at">p.value =</span> <span class="fl">0.2</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames    start      end width strand  source type
ENSMUSG00000093185     chr4 45396994 45397118   125      - ENSEMBL gene
ENSMUSG00000004328     chr7 17031507 17062427 30921      -  HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000093185       125    NA ENSMUSG00000093185          miRNA   Gm25692
ENSMUSG00000004328      6710    NA ENSMUSG00000004328 protein_coding     Hif3a
                   level      mgi_id          havana_gene  tag     logFC
ENSMUSG00000093185     3 MGI:5455469                 &lt;NA&gt; &lt;NA&gt; 0.9495883
ENSMUSG00000004328     2 MGI:1859778 OTTMUSG00000022817.2 &lt;NA&gt; 1.2763016
                     AveExpr        t      P.Value adj.P.Val        B
ENSMUSG00000093185 0.9472638 5.038308 1.037956e-05 0.1214346 3.264081
ENSMUSG00000004328 2.3421403 5.028972 1.069390e-05 0.1214346 3.312618</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Retina"</span>, <span class="at">p.value =</span> <span class="fl">0.2</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames    start      end width strand source type
ENSMUSG00000050440     chr7 30942368 30944032  1665      - HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000050440       410    NA ENSMUSG00000050440 protein_coding      Hamp
                   level      mgi_id          havana_gene  tag     logFC
ENSMUSG00000050440     2 MGI:1933533 OTTMUSG00000058034.2 &lt;NA&gt; -3.983747
                     AveExpr         t      P.Value  adj.P.Val         B
ENSMUSG00000050440 -5.182209 -5.515318 2.238313e-06 0.05083434 -2.172283</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Cortex"</span>, <span class="at">p.value =</span> <span class="fl">0.2</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)<span class="sc">$</span>gene_id,</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Hippocampus"</span>, <span class="at">p.value =</span> <span class="fl">0.2</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)<span class="sc">$</span>gene_id,</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Retina"</span>, <span class="at">p.value =</span> <span class="fl">0.2</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)<span class="sc">$</span>gene_id)</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm[to_plot, ] <span class="sc">|&gt;</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>group_no_tissue <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">paste0</span>(df<span class="sc">$</span>treatment, <span class="st">"."</span>, df<span class="sc">$</span>sex), <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Vehicle.female"</span>,</span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tamoxifen.female"</span>, <span class="st">"Vehicle.male"</span>, <span class="st">"Tamoxifen.male"</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"V.f"</span>, <span class="st">"T.f"</span>,</span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"V.m"</span>, <span class="st">"T.m"</span>))</span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>group_gene_tissue <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df<span class="sc">$</span>gene_name, <span class="st">" in "</span>, df<span class="sc">$</span>tissue)</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group_no_tissue, <span class="at">y =</span> log2CPM, <span class="at">col =</span> group_no_tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>group_gene_tissue,</span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">ncol =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="hands_on_limma_advanced_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> How would you describe the behavior of these genes across conditions and tissues, in (well chosen) simple words? Are these genes described in the original paper?</p>
</section>
</section>
<section id="gsea" class="level1">
<h1>GSEA</h1>
<p>It is possible that we detect few significant genes for the interaction but that some trends are visible collectively at the gene set level</p>
<div class="cell" data-hash="hands_on_limma_advanced_cache/html/unnamed-chunk-10_d8c36b37f82adf4d7683fe7eb158bcd2">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>GOmapping <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Mm.eg.db, <span class="at">keys =</span> dge<span class="sc">$</span>genes<span class="sc">$</span>gene_id, <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GOALL"</span>, <span class="st">"EVIDENCEALL"</span>, <span class="st">"ONTOLOGYALL"</span>))</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Transform to list</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>GOmapping <span class="ot">&lt;-</span> <span class="fu">tapply</span>(GOmapping<span class="sc">$</span>ENSEMBL, <span class="fu">list</span>(GOmapping<span class="sc">$</span>GOALL), unique)</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a><span class="do">## process and filter gene sets</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>indx <span class="ot">&lt;-</span> <span class="fu">ids2indices</span>(<span class="at">gene.sets =</span> GOmapping, <span class="at">identifiers =</span> dge<span class="sc">$</span>genes<span class="sc">$</span>gene_id)</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter gene sets smaller than 10 genes, and gene sets larger than 500 genes</span></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>indx <span class="ot">&lt;-</span> indx[<span class="fu">sapply</span>(indx, length) <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="sc">&amp;</span> <span class="fu">sapply</span>(indx, length) <span class="sc">&lt;</span> <span class="dv">500</span>]</span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Run CameraPR on the t-statistics Cortex</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a><span class="do">## ######################################################################</span></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Cortex"</span>, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">n =</span> <span class="cn">Inf</span>, <span class="at">sort.by =</span> <span class="st">"none"</span>)</span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cameraPR</span>(<span class="at">statistic =</span> top<span class="sc">$</span>t, <span class="at">index =</span> indx)</span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Add info on Gene Ontology terms</span></span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cbind</span>(res_gsea, AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(GO.db, <span class="at">keys =</span> <span class="fu">row.names</span>(res_gsea),</span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype =</span> <span class="st">"GOID"</span>, <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"ONTOLOGY"</span>, <span class="st">"TERM"</span>)))  <span class="co">#, 'DEFINITION')))</span></span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(res_gsea<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 8057     9 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_gsea, <span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           NGenes Direction       PValue          FDR       GOID ONTOLOGY
GO:0009584     33      Down 4.328433e-09 3.491314e-05 GO:0009584       BP
GO:0009583     53      Down 1.158079e-08 4.670531e-05 GO:0009583       BP
GO:0042622     14      Down 3.113827e-08 8.372043e-05 GO:0042622       CC
GO:0001750     84      Down 1.496167e-07 3.017021e-04 GO:0001750       CC
GO:0050908     24      Down 6.373957e-06 8.568722e-03 GO:0050908       BP
GO:0050962     24      Down 6.373957e-06 8.568722e-03 GO:0050962       BP
GO:0007602     29      Down 1.716819e-05 1.978266e-02 GO:0007602       BP
GO:0050953    149      Down 4.241912e-05 4.276907e-02 GO:0050953       BP
GO:0007601    146      Down 4.988848e-05 4.471116e-02 GO:0007601       BP
GO:0001917     59      Down 2.470557e-04 1.992751e-01 GO:0001917       CC
                                                                 TERM
GO:0009584                                 detection of visible light
GO:0009583                                detection of light stimulus
GO:0042622                       photoreceptor outer segment membrane
GO:0001750                                photoreceptor outer segment
GO:0050908  detection of light stimulus involved in visual perception
GO:0050962 detection of light stimulus involved in sensory perception
GO:0007602                                          phototransduction
GO:0050953                       sensory perception of light stimulus
GO:0007601                                          visual perception
GO:0001917                                photoreceptor inner segment</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Hippocampus</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="do">## ######################################################################</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Hippocampus"</span>, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">n =</span> <span class="cn">Inf</span>, <span class="at">sort.by =</span> <span class="st">"none"</span>)</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cameraPR</span>(<span class="at">statistic =</span> top<span class="sc">$</span>t, <span class="at">index =</span> indx)</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cbind</span>(res_gsea, AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(GO.db, <span class="at">keys =</span> <span class="fu">row.names</span>(res_gsea),</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype =</span> <span class="st">"GOID"</span>, <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"ONTOLOGY"</span>, <span class="st">"TERM"</span>)))  <span class="co">#, 'DEFINITION')))</span></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(res_gsea<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 8047    19 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_gsea, <span class="at">n =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           NGenes Direction       PValue        FDR       GOID ONTOLOGY
GO:0099563     42        Up 2.188246e-06 0.01062105 GO:0099563       BP
GO:0099010     34        Up 2.633537e-06 0.01062105 GO:0099010       BP
GO:0098885     20        Up 2.606091e-05 0.05108872 GO:0098885       BP
GO:1990089     44        Up 3.166918e-05 0.05108872 GO:1990089       BP
GO:1990090     44        Up 3.166918e-05 0.05108872 GO:1990090       BP
GO:0046875     32        Up 5.060574e-05 0.06483732 GO:0046875       MF
GO:0031209     12        Up 7.092415e-05 0.06483732 GO:0031209       CC
GO:0051861     25        Up 7.182758e-05 0.06483732 GO:0051861       MF
GO:0016514     28        Up 7.345699e-05 0.06483732 GO:0016514       CC
GO:0043620     41        Up 9.048976e-05 0.06483732 GO:0043620       BP
GO:0099186     25        Up 9.608726e-05 0.06483732 GO:0099186       MF
GO:0002931     46        Up 9.646018e-05 0.06483732 GO:0002931       BP
GO:0090208     22        Up 1.378875e-04 0.08449534 GO:0090208       BP
GO:0098685    143        Up 1.488151e-04 0.08449534 GO:0098685       CC
GO:0140288     14        Up 1.571324e-04 0.08449534 GO:0140288       CC
GO:0048671     10        Up 1.691701e-04 0.08528286 GO:0048671       BP
GO:0043618     36        Up 2.019481e-04 0.09581842 GO:0043618       BP
GO:0038179     41        Up 2.216341e-04 0.09848957 GO:0038179       BP
GO:0035254     76        Up 2.319987e-04 0.09848957 GO:0035254       MF
GO:0098918     35        Up 2.700314e-04 0.10442186 GO:0098918       MF
                                                                                        TERM
GO:0099563                                                modification of synaptic structure
GO:0099010                                            modification of postsynaptic structure
GO:0098885                                   modification of postsynaptic actin cytoskeleton
GO:1990089                                                   response to nerve growth factor
GO:1990090                                 cellular response to nerve growth factor stimulus
GO:0046875                                                           ephrin receptor binding
GO:0031209                                                                      SCAR complex
GO:0051861                                                                glycolipid binding
GO:0016514                                                                   SWI/SNF complex
GO:0043620                   regulation of DNA-templated transcription in response to stress
GO:0099186                                             structural constituent of postsynapse
GO:0002931                                                              response to ischemia
GO:0090208                             positive regulation of triglyceride metabolic process
GO:0098685                                                 Schaffer collateral - CA1 synapse
GO:0140288                                                                      GBAF complex
GO:0048671                                       negative regulation of collateral sprouting
GO:0043618 regulation of transcription from RNA polymerase II promoter in response to stress
GO:0038179                                                    neurotrophin signaling pathway
GO:0035254                                                        glutamate receptor binding
GO:0098918                                                 structural constituent of synapse</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Retina</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="do">## ######################################################################</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction_Retina"</span>, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">n =</span> <span class="cn">Inf</span>, <span class="at">sort.by =</span> <span class="st">"none"</span>)</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cameraPR</span>(<span class="at">statistic =</span> top<span class="sc">$</span>t, <span class="at">index =</span> indx)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cbind</span>(res_gsea, AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(GO.db, <span class="at">keys =</span> <span class="fu">row.names</span>(res_gsea),</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype =</span> <span class="st">"GOID"</span>, <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"ONTOLOGY"</span>, <span class="st">"TERM"</span>)))  <span class="co">#, 'DEFINITION')))</span></span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(res_gsea<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE 
 8066 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_gsea, <span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           NGenes Direction       PValue       FDR       GOID ONTOLOGY
GO:0005684     98      Down 9.373433e-05 0.6548716 GO:0005684       CC
GO:0071005     53      Down 2.142637e-04 0.6548716 GO:0071005       CC
GO:0071011     55      Down 2.653025e-04 0.6548716 GO:0071011       CC
GO:0005839     19      Down 4.145910e-04 0.6548716 GO:0005839       CC
GO:0005201    128        Up 4.279018e-04 0.6548716 GO:0005201       MF
GO:0030318     28        Up 6.431946e-04 0.6548716 GO:0030318       BP
GO:0071007     34      Down 1.063285e-03 0.6548716 GO:0071007       CC
GO:0070262     12        Up 1.095004e-03 0.6548716 GO:0070262       BP
GO:0015020     23        Up 1.156632e-03 0.6548716 GO:0015020       MF
GO:0070888     46        Up 1.195412e-03 0.6548716 GO:0070888       MF
                                                  TERM
GO:0005684                U2-type spliceosomal complex
GO:0071005            U2-type precatalytic spliceosome
GO:0071011                    precatalytic spliceosome
GO:0005839                     proteasome core complex
GO:0005201 extracellular matrix structural constituent
GO:0030318                  melanocyte differentiation
GO:0071007        U2-type catalytic step 2 spliceosome
GO:0070262           peptidyl-serine dephosphorylation
GO:0015020            glucuronosyltransferase activity
GO:0070888                               E-box binding</code></pre>
</div>
</div>
</section>
<section id="to-finish" class="level1">
<h1>To finish</h1>
<ul>
<li><p><em>“Interpretations of sex-related variation are not always commensurate with the story the data actually tell”</em> (Pape et al.&nbsp;2024). How do you judge the interpretations of the results in the published paper?</p></li>
<li><p>Discuss potential limitations of the paper and the conclusion about the absence of Tamoxifen effect.</p>
<ul>
<li>What is important to notice in the design of the experiment (tamoxifen treatment duration, batches, number of replicates, …)?</li>
<li>Have a look at the results and materials and methods section: which analyses were made and how could they influence the results?</li>
</ul></li>
<li><p>Related to our reanalysis: what is improved here by separating the analysos tissue-by-tissue?</p></li>
<li><p><em>“Sex is not a causal mechanism”</em> (Pape et al.&nbsp;2024): follow-up on some DE genes and try to find which mechanism of action could explain their differential expression patterns.</p></li>
<li><p>Discuss what would be the next steps if that was your research project. Which experiments would you design next? How would you describe the results in a paper?</p>
<ul>
<li>You can also have a look at other related papers (e.g., https://www.nature.com/articles/s41586-022-04686-1, https://www.ahajournals.org/doi/10.1161/ATVBAHA.123.319922)</li>
<li>Or look at follow-up papers from the same authors (e.g., https://link.springer.com/article/10.1007/s12035-022-02860-0)</li>
</ul></li>
</ul>
<p><sub>Icons taken from http://www.flaticon.com/</sub></p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>